diff -ru dde-grand-search-5.4.5.orig/src/preview-plugin/audio-preview/audiofileinfo.cpp dde-grand-search-5.4.5/src/preview-plugin/audio-preview/audiofileinfo.cpp
--- dde-grand-search-5.4.5.orig/src/preview-plugin/audio-preview/audiofileinfo.cpp	2024-01-24 20:16:45.546539948 +0100
+++ dde-grand-search-5.4.5/src/preview-plugin/audio-preview/audiofileinfo.cpp	2024-01-24 20:17:21.294540575 +0100
@@ -54,9 +54,9 @@
 {
     TagLib::Tag *tag = static_cast<TagLib::Tag *>(obj);
     bool encode = true;
-    encode &= tag->title().isNull() ? true : tag->title().isLatin1();
-    encode &= tag->artist().isNull() ? true : tag->artist().isLatin1();
-    encode &= tag->album().isNull() ? true : tag->album().isLatin1();
+    encode &= tag->title().isEmpty() ? true : tag->title().isLatin1();
+    encode &= tag->artist().isEmpty() ? true : tag->artist().isLatin1();
+    encode &= tag->album().isEmpty() ? true : tag->album().isLatin1();
 
     QByteArray detectByte;
     QByteArray detectCodec;
