--- a/build/update_pkg.sh	2024-01-10 14:06:25.000000000 +0800
+++ b/build/update_pkg.sh	2024-01-10 14:31:06.474082272 +0800
@@ -25,7 +25,7 @@
 echo install gocode ...
 go install -v github.com/visualfc/gocode@latest
 echo install gotools ...
-go install -v github.com/visualfc/gotools@latest
+go install -v github.com/visualfc/gotools@master
 echo install gomodifytags ...
 go install -v github.com/fatih/gomodifytags@latest
 
