diff --git a/samples/vst-hosting/inspectorapp/CMakeLists.txt b/samples/vst-hosting/inspectorapp/CMakeLists.txt
index ed6eaa8..f3b6d83 100644
--- a/samples/vst-hosting/inspectorapp/CMakeLists.txt
+++ b/samples/vst-hosting/inspectorapp/CMakeLists.txt
@@ -52,7 +52,7 @@ if(TARGET vstgui AND TARGET vstgui_standalone)
     if(NOT SMTG_MAC)
         set_target_properties(${target}
             PROPERTIES
-                RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<$<CONFIG:Debug>:bin/Debug>$<$<CONFIG:Release>:bin/Release>$<$<CONFIG:ReleaseLTO>:bin/ReleaseLTO>/VST3Inspector
+                RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<$<CONFIG:Debug>:bin/Debug>$<$<CONFIG:Release>:bin/Release>$<$<CONFIG:ReleaseLTO>:bin/ReleaseLTO>$<$<CONFIG:None>:${CMAKE_BINARY_DIR}/bin/None/>/VST3Inspector
         )
     endif(NOT SMTG_MAC)
 
