diff --color --unified --recursive --text src/vst3sdk/public.sdk/samples/vst-hosting/inspectorapp/CMakeLists.txt package.new/vst3sdk/public.sdk/samples/vst-hosting/inspectorapp/CMakeLists.txt
--- src/vst3sdk/public.sdk/samples/vst-hosting/inspectorapp/CMakeLists.txt	2023-07-28 11:17:15.331067137 +0200
+++ package.new/vst3sdk/public.sdk/samples/vst-hosting/inspectorapp/CMakeLists.txt	2023-07-28 11:16:53.154300021 +0200
@@ -52,7 +52,7 @@
     if(NOT SMTG_MAC)
         set_target_properties(${target}
             PROPERTIES
-                RUNTIME_OUTPUT_DIRECTORY $<$<CONFIG:Debug>:${CMAKE_BINARY_DIR}/bin/Debug/>$<$<CONFIG:Release>:${CMAKE_BINARY_DIR}/bin/Release/>$<$<CONFIG:ReleaseLTO>:${CMAKE_BINARY_DIR}/bin/ReleaseLTO/>/VST3Inspector
+                RUNTIME_OUTPUT_DIRECTORY $<$<CONFIG:Debug>:${CMAKE_BINARY_DIR}/bin/Debug/>$<$<CONFIG:Release>:${CMAKE_BINARY_DIR}/bin/Release/>$<$<CONFIG:ReleaseLTO>:${CMAKE_BINARY_DIR}/bin/ReleaseLTO/>$<$<CONFIG:None>:${CMAKE_BINARY_DIR}/bin/None/>/VST3Inspector
         )
     endif(NOT SMTG_MAC)
 
