Index: fakechroot-2.20.1/src/libfakechroot.h
===================================================================
--- fakechroot-2.20.1.orig/src/libfakechroot.h
+++ fakechroot-2.20.1/src/libfakechroot.h
@@ -32,6 +32,9 @@
 
 #define debug fakechroot_debug
 
+#ifdef __loongarch64
+#define _STAT_VER 0
+#endif
 
 #ifdef HAVE___ATTRIBUTE__VISIBILITY
 # define LOCAL __attribute__((visibility("hidden")))
@@ -200,7 +203,6 @@
 # endif
 #endif
 
-
 typedef void (*fakechroot_wrapperfn_t)(void);
 
 struct fakechroot_wrapper {
