Index: open-iscsi-2.1.6/utils/fwparam_ibft/fwparam.h
===================================================================
--- open-iscsi-2.1.6.orig/utils/fwparam_ibft/fwparam.h
+++ open-iscsi-2.1.6/utils/fwparam_ibft/fwparam.h
@@ -17,7 +17,6 @@
 #ifndef FWPARAM_H_
 #define FWPARAM_H_
 
-#include <stdint.h>
 #include "fw_context.h"
 
 #define FILENAMESZ (1024)
Index: open-iscsi-2.1.6/include/iscsi_proto.h
===================================================================
--- open-iscsi-2.1.6.orig/include/iscsi_proto.h
+++ open-iscsi-2.1.6/include/iscsi_proto.h
@@ -63,7 +63,7 @@
 #endif
 
 /* initiator tags; opaque for target */
-typedef uint32_t __bitwise__ itt_t;
+typedef uint32_t itt_t;
 /* below makes sense only for initiator that created this tag */
 #define build_itt(itt, age) ((__force itt_t)\
 	((itt) | ((age) << ISCSI_AGE_SHIFT)))
