diff --git a/pyproject.toml b/pyproject.toml
index e1b7122..459c2a3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,9 +18,6 @@ keywords = []
 dependencies = [
     # Ensure numpy release supports Python version.
     "numpy>1.20",
-    "numpy>=1.21.2; python_version>='3.10'",
-    "numpy>=1.23.3; python_version>='3.11'",
-    "numpy>=1.26.0; python_version>='3.12'",
 ]
 
 [project.urls]
@@ -49,10 +46,9 @@ pyink-use-majority-quotes = true
 [build-system]
 requires = [
     # Build with oldest supported numpy for each Python version.
-    "numpy~=1.21.2; python_version<'3.11'",
-    "numpy~=1.23.3; python_version>='3.11' and python_version<'3.12'",
-    "numpy~=1.26.0; python_version>='3.12'",
-    "setuptools~=68.1.0",
+    "numpy>1.20",
+    "pybind11>2.10.0",
+    "setuptools>68.1.0",
 ]
 build-backend = "setuptools.build_meta"
 
