diff -aur pybox2d-2.3.10.orig/library/Box2D/tests/test_joints.py pybox2d-2.3.10/library/Box2D/tests/test_joints.py
--- pybox2d-2.3.10.orig/library/Box2D/tests/test_joints.py	2024-04-13 20:23:36.783792892 +0200
+++ pybox2d-2.3.10/library/Box2D/tests/test_joints.py	2024-04-13 20:23:58.940927282 +0200
@@ -97,7 +97,7 @@
         else:
             b = getattr(joint, prop)
 
-        self.assertEquals(a, b, "Property not equal from definition to joint: %s (dfn %s != joint %s)" % (prop, a, b) )
+        self.assertEqual(a, b, "Property not equal from definition to joint: %s (dfn %s != joint %s)" % (prop, a, b) )
 
     # ---- revolute joint ----
     def revolute_definition(self, body1, body2, anchor):
