Index: Firebird-3.0.7.33374-0/src/common/classes/DbImplementation.cpp
===================================================================
--- Firebird-3.0.7.33374-0.orig/src/common/classes/DbImplementation.cpp
+++ Firebird-3.0.7.33374-0/src/common/classes/DbImplementation.cpp
@@ -49,6 +49,7 @@ static const UCHAR CpuAlpha = 14;
 static const UCHAR CpuArm64 = 15;
 static const UCHAR CpuPowerPc64el = 16;
 static const UCHAR CpuM68k = 17;
+static const UCHAR CpuLoongarch64 = 18;
 
 static const UCHAR OsWindows = 0;
 static const UCHAR OsLinux = 1;
Index: Firebird-3.0.7.33374-0/src/common/common.h
===================================================================
--- Firebird-3.0.7.33374-0.orig/src/common/common.h
+++ Firebird-3.0.7.33374-0/src/common/common.h
@@ -140,6 +140,10 @@
 #define RISC_ALIGNMENT
 #endif /* sparc */
 
+#ifdef _LOONGARCH_ARCH
+#define FB_CPU CpuLoongarch64
+#endif /* loongarch64 */
+
 #ifdef MIPSEL
 #define FB_CPU CpuMipsel
 #endif /* mipsel */
