Index: box2d-2.4.1/unit-test/doctest.h
===================================================================
--- box2d-2.4.1.orig/unit-test/doctest.h
+++ box2d-2.4.1/unit-test/doctest.h
@@ -4018,7 +4018,7 @@ namespace {
         static bool             isSet;
         static struct sigaction oldSigActions[DOCTEST_COUNTOF(signalDefs)];
         static stack_t          oldSigStack;
-        static char             altStackMem[4 * SIGSTKSZ];
+        static char             altStackMem[65536];
 
         static void handleSignal(int sig) {
             const char* name = "<unknown signal>";
