Index: mkinitcpio-archiso/hooks/archiso
===================================================================
--- mkinitcpio-archiso.orig/hooks/archiso
+++ mkinitcpio-archiso/hooks/archiso
@@ -170,6 +170,7 @@ _verify_cms_signature() {
 
 run_hook() {
     [ -z "${arch}" ] && arch="$(uname -m)"
+    [ ${arch} == "loongarch64" ] && arch=loong64
     [ -z "${copytoram}" ] && copytoram="auto"
     [ -z "${copytoram_size}" ] && copytoram_size="75%"
     [ -z "${archisobasedir}" ] && archisobasedir="arch"
