Index: mkinitcpio-archiso/hooks/archiso
===================================================================
--- mkinitcpio-archiso.orig/hooks/archiso
+++ mkinitcpio-archiso/hooks/archiso
@@ -136,6 +136,9 @@ _verify_signature() {
 
 run_hook() {
     [ -z "${arch}" ] && arch="$(uname -m)"
+    if [[ ${arch} == "loongarch64" ]];then
+        arch=loong64
+    fi
     [ -z "${copytoram_size}" ] && copytoram_size="75%"
     [ -z "${archisobasedir}" ] && archisobasedir="arch"
     [ -z "${dm_snap_prefix}" ] && dm_snap_prefix="arch"
