From dda3c4bfdf9010e53c6e0259bc11c43922e51398 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Tue, 28 Nov 2023 22:12:58 +0800
Subject: [PATCH] add loong64 support

---
 repod/common/enums.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/repod/common/enums.py b/repod/common/enums.py
index 64dc74b..773fd81 100644
--- a/repod/common/enums.py
+++ b/repod/common/enums.py
@@ -23,6 +23,8 @@ class ArchitectureEnum(Enum):
         The i486 CPU architecture
     I686: "i686"
         The i686 CPU architecture
+    LOONG64: "loong64"
+        The loong64 CPU architecture
     PENTIUM4: "pentium4"
         The pentium4 CPU architecture
     RISCV32: "riscv32"
@@ -46,6 +48,7 @@ class ArchitectureEnum(Enum):
     ARMV7H = "armv7h"
     I486 = "i486"
     I686 = "i686"
+    LOONG64 = "loong64"
     PENTIUM4 = "pentium4"
     RISCV32 = "riscv32"
     RISCV64 = "riscv64"
-- 
2.42.0

