From eb6dd0fff6be1f9afc7b660cebd898c36df4c6d0 Mon Sep 17 00:00:00 2001
From: Ex-32 <jenna@fligor.net>
Date: Fri, 9 Jun 2023 18:50:24 -0500
Subject: [PATCH] fixed terminal frame issue with nvidia drivers

---
 app/qml/TerminalFrame.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/qml/TerminalFrame.qml b/app/qml/TerminalFrame.qml
index 3e1baef..49baa2c 100644
--- a/app/qml/TerminalFrame.qml
+++ b/app/qml/TerminalFrame.qml
@@ -87,7 +87,7 @@ ShaderEffect {
             float alpha = 0.0;
 
             float frameShadow = max2(positiveLog(-coords * frameShadowCoeff + vec2(1.0)) + positiveLog(coords * frameShadowCoeff - (vec2(frameShadowCoeff) - vec2(1.0))));
-            frameShadow = max(sqrt(frameShadow), 0.0);
+            frameShadow = clamp(sqrt(frameShadow), 0.0, 1.0);
             color *= frameShadow;
             alpha = sum2(1.0 - step(vec2(0.0), coords) + step(vec2(1.0), coords));
             alpha = clamp(alpha, 0.0, 1.0);
-- 
2.41.0

