Index: deepin-desktop-base-2021.06.16/Makefile
===================================================================
--- deepin-desktop-base-2021.06.16.orig/Makefile
+++ deepin-desktop-base-2021.06.16/Makefile
@@ -23,6 +23,8 @@ build:
 		sed -e "s|@@VERSION@@|$(VERSION)|g" -e "s|@@RELEASE@@|$(RELEASE)|g" files/os-version-loongson > files/os-version
         else ifeq (${ARCH_BUILD}, mips64)
 		sed -e "s|@@VERSION@@|$(VERSION)|g" -e "s|@@RELEASE@@|$(RELEASE)|g" files/desktop-version-loongson.in > files/desktop-version
+        else ifeq (${ARCH_BUILD}, loongarch64)
+		sed -e "s|@@VERSION@@|$(VERSION)|g" -e "s|@@RELEASE@@|$(RELEASE)|g" files/desktop-version-loongarch64.in > files/desktop-version
         else ifeq (${ARCH_BUILD}, sw_64)
 		sed -e "s|@@VERSION@@|$(VERSION)|g" -e "s|@@RELEASE@@|$(RELEASE)|g" files/desktop-version-sw.in > files/desktop-version
 		sed -e "s|@@VERSION@@|$(VERSION)|g" -e "s|@@RELEASE@@|$(RELEASE)|g" files/os-version-sw > files/os-version
Index: deepin-desktop-base-2021.06.16/files/desktop-version-loongarch64.in
===================================================================
--- /dev/null
+++ deepin-desktop-base-2021.06.16/files/desktop-version-loongarch64.in
@@ -0,0 +1,9 @@
+[Release]
+Version=@@VERSION@@
+Type=Desktop
+Type[zh_CN]=社区版
+Edition=Y2020E0001
+Copyright=Y2020CR001
+[Addition]
+Milestone=@@RELEASE@@
+Buildid=build1
