From 2ca4d5f5288a904e4c8208c74537cb7d59469f24 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Sun, 18 Apr 2021 22:37:24 +0800
Subject: [PATCH] just for la64 build

---
 src/miniacc.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/miniacc.h b/src/miniacc.h
index 3c949280..e52cda58 100644
--- a/src/miniacc.h
+++ b/src/miniacc.h
@@ -979,6 +979,9 @@
 #elif (UINT_MAX == ACC_0xffffL) && defined(__m32c__)
 #  define ACC_ARCH_M16C             1
 #  define ACC_INFO_ARCH             "m16c"
+#elif defined(__loongarch64)
+#  define ACC_ARCH_LA64             1
+#  define ACC_INFO_ARCH             "la64"
 #elif defined(__IAR_SYSTEMS_ICC__) && defined(__ICCM16C__)
 #  define ACC_ARCH_M16C             1
 #  define ACC_INFO_ARCH             "m16c"
@@ -2586,6 +2589,8 @@ ACC_COMPILE_TIME_ASSERT_HEADER(ACC_SIZEOF_PTRDIFF_T == sizeof(ptrdiff_t))
 #  define ACC_ABI_BIG_ENDIAN        1
 #elif 1 && (ACC_ARCH_ARM64) && defined(__AARCH64EL__) && !defined(__AARCH64EB__)
 #  define ACC_ABI_LITTLE_ENDIAN     1
+#elif 1 && (ACC_ARCH_LA64) && defined(__loongarch64)
+#  define ACC_ABI_LITTLE_ENDIAN     1
 #elif 1 && (ACC_ARCH_ARM64) && defined(_MSC_VER) && defined(_WIN32)
 #  define ACC_ABI_LITTLE_ENDIAN     1
 #elif 1 && (ACC_ARCH_MIPS) && defined(__MIPSEB__) && !defined(__MIPSEL__)
-- 
2.28.0

