diff --git a/reader/sidebar/BookMarkWidget.cpp b/reader/sidebar/BookMarkWidget.cpp
index e2aa143d..b30d66a4 100644
--- a/reader/sidebar/BookMarkWidget.cpp
+++ b/reader/sidebar/BookMarkWidget.cpp
@@ -48,7 +48,7 @@ BookMarkWidget::~BookMarkWidget()
 
 void BookMarkWidget::initWidget()
 {
-    connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
+    connect(Dtk::Gui::DGuiApplicationHelper::instance(), &Dtk::Gui::DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
 
     m_pImageListView = new SideBarImageListView(m_sheet, this);
     m_pImageListView->setAccessibleName("View_ImageList");
@@ -205,7 +205,7 @@ void BookMarkWidget::showMenu()
 
 void BookMarkWidget::onUpdateTheme()
 {
-    DPalette plt = Dtk::Gui::DGuiApplicationHelper::instance()->applicationPalette();
+    Dtk::Gui::DPalette plt = Dtk::Gui::DGuiApplicationHelper::instance()->applicationPalette();
     plt.setColor(Dtk::Gui::DPalette::Background, plt.color(Dtk::Gui::DPalette::Base));
     setPalette(plt);
 
