--- supermin-5.3.2/configure.ac	2024-01-30 11:51:10.000000000 +0800
+++ supermin-5.3.2/configure.ac	2024-01-30 14:20:02.169340385 +0800
@@ -114,7 +114,7 @@
 if test "x$PERLDOC" = "xno" ; then
   AC_MSG_WARN([perldoc not found - install perl to make man pages])
 fi
-AM_CONDITIONAL(HAVE_PERLDOC,[test "x$PERLDOC" != "xno"])
+AM_CONDITIONAL(HAVE_PERLDOC,[test "x$PERLDOC" != "x_no"])
 
 dnl For yum-rpm handler.
 AC_PATH_PROG(RPM,[rpm],[no])
