Index: postgresql-13.6/src/include/storage/s_lock.h
===================================================================
--- postgresql-13.6.orig/src/include/storage/s_lock.h
+++ postgresql-13.6/src/include/storage/s_lock.h
@@ -321,7 +321,7 @@
  * We use the int-width variant of the builtin because it works on more chips
  * than other widths.
  */
-#if defined(__arm__) || defined(__arm) || defined(__aarch64__) || defined(__aarch64)
+#if defined(__arm__) || defined(__arm) || defined(__aarch64__) || defined(__aarch64) || defined(__loongarch64)
 #ifdef HAVE_GCC__SYNC_INT32_TAS
 #define HAS_TEST_AND_SET
 
