Index: postgresql-13.6/src/include/storage/s_lock.h
===================================================================
--- postgresql-13.6.orig/src/include/storage/s_lock.h
+++ postgresql-13.6/src/include/storage/s_lock.h
@@ -342,9 +342,9 @@ tas(volatile slock_t *lock)
 
 
 /*
- * RISC-V likewise uses __sync_lock_test_and_set(int *, int) if available.
+ * RISC-V and LoongArch likewise uses __sync_lock_test_and_set(int *, int) if available.
  */
-#if defined(__riscv)
+#if defined(__riscv) || defined(__loongarch__)
 #ifdef HAVE_GCC__SYNC_INT32_TAS
 #define HAS_TEST_AND_SET
 
