diff --git a/i18n_ts/CMakeLists.txt b/i18n_ts/CMakeLists.txt
index f4b0cef..a421287 100644
--- a/i18n_ts/CMakeLists.txt
+++ b/i18n_ts/CMakeLists.txt
@@ -1,7 +1,7 @@
 find_package(Qt5LinguistTools)
 
 file(GLOB ts_files *.ts)
-qt5_create_translation(qm_files ${ts_files})
+qt5_create_translation(qm_files ${CMAKE_CURRENT_SOURCE_DIR} ${ts_files})
 add_custom_target(i18n
 	DEPENDS ${qm_files}
 	SOURCES ${ts_files}
