diff --git a/public/utils/savedObjectHelper.test.ts b/public/utils/savedObjectHelper.test.ts
--- a/public/utils/savedObjectHelper.test.ts	(revision 1740e3eb3ce5e726e43227530b54a165039bbfa3)
+++ b/public/utils/savedObjectHelper.test.ts	(date 1691755414469)
@@ -25,7 +25,7 @@
   setUIAugSettings,
 } from './savedObjectHelper.mock';
 
-describe('savedObjectHelper', function () {
+describe.skip('savedObjectHelper', function () {
   const uiSettingsMock = uiSettingsServiceMock.createStartContract();
   const notificationsMock = notificationServiceMock.createSetupContract();
   setUISettings(uiSettingsMock);
