diff --git a/src/libxml2_loader.cpp b/src/libxml2_loader.cpp
index af6104bff..8191d9e08 100644
--- a/src/libxml2_loader.cpp
+++ b/src/libxml2_loader.cpp
@@ -79,7 +79,7 @@ public:
 
         if (!doc)
         {
-            xmlError * error = xmlCtxtGetLastError(ctx_);
+            const xmlError * error = xmlCtxtGetLastError(ctx_);
             if (error)
             {
                 std::string msg("XML document not well formed:\n");
@@ -118,7 +118,7 @@ public:
         if (!doc)
         {
             std::string msg("XML document not well formed");
-            xmlError * error = xmlCtxtGetLastError( ctx_ );
+            const xmlError * error = xmlCtxtGetLastError( ctx_ );
             if (error)
             {
                 msg += ":\n";
