diff -aur librepo-1.17.0.old/CMakeLists.txt librepo-1.17.0/CMakeLists.txt
--- librepo-1.17.0.old/CMakeLists.txt	2024-03-25 21:20:16.874681785 +0100
+++ librepo-1.17.0/CMakeLists.txt	2024-03-25 21:20:42.054785557 +0100
@@ -37,7 +37,7 @@
 FIND_PACKAGE(CURL 7.52.0 REQUIRED)
 
 IF (USE_GPGME)
-    FIND_PACKAGE(Gpgme REQUIRED)
+    PKG_SEARCH_MODULE(GPGME REQUIRED gpgme)
     IF (ENABLE_SELINUX)
         PKG_CHECK_MODULES(SELINUX REQUIRED libselinux)
     ENDIF(ENABLE_SELINUX)
diff -aur librepo-1.17.0.old/librepo/CMakeLists.txt librepo-1.17.0/librepo/CMakeLists.txt
--- librepo-1.17.0.old/librepo/CMakeLists.txt	2024-03-25 21:20:16.874681785 +0100
+++ librepo-1.17.0/librepo/CMakeLists.txt	2024-03-25 21:21:00.624861703 +0100
@@ -70,7 +70,7 @@
                         ${GLIB2_LIBRARIES}
                      )
 IF (USE_GPGME)
-    TARGET_LINK_LIBRARIES(librepo ${GPGME_VANILLA_LIBRARIES})
+    TARGET_LINK_LIBRARIES(librepo ${GPGME_LIBRARIES})
     IF (ENABLE_SELINUX)
         TARGET_LINK_LIBRARIES(librepo ${SELINUX_LIBRARIES})
     ENDIF(ENABLE_SELINUX)
