diff --git a/src/widgets/dkeysequenceedit.cpp b/src/widgets/dkeysequenceedit.cpp
index a99e17ae..b3e34014 100644
--- a/src/widgets/dkeysequenceedit.cpp
+++ b/src/widgets/dkeysequenceedit.cpp
@@ -276,7 +276,7 @@ void DKeySequenceEdit::keyPressEvent(QKeyEvent *e)
     }
 
     if (e->modifiers() & Qt::ShiftModifier) {
-        QList<int> possibleKeys = QKeyMapper::possibleKeys(e);
+        QList<QKeyCombination> possibleKeys = QKeyMapper::possibleKeys(e);
         int pkTotal = possibleKeys.count();
         if (!pkTotal)
             return;
