diff -aur feeluown-4.1.2.old/feeluown/player/lyric.py feeluown-4.1.2/feeluown/player/lyric.py
--- feeluown-4.1.2.old/feeluown/player/lyric.py	2024-04-09 19:41:15.895337894 +0200
+++ feeluown-4.1.2/feeluown/player/lyric.py	2024-04-09 19:46:03.386175235 +0200
@@ -50,9 +50,9 @@
     Reference: https://github.com/osdlyrics/osdlyrics/blob/master/python/lrc.py
 
     >>> parse_lyric_text("[00:00.00] 作曲 : 周杰伦\\n[00:01.00] 作词 : 周杰伦\\n")
-    OrderedDict([(0, ' 作曲 : 周杰伦'), (1000, ' 作词 : 周杰伦')])
+    OrderedDict({0, ' 作曲 : 周杰伦', 1000, ' 作词 : 周杰伦'})
     >>> parse_lyric_text("[01:30][00:01:10][01:00]再等直至再吻到你")
-    OrderedDict([(60000, '再等直至再吻到你'), (70000, '再等直至再吻到你'), (90000, '再等直至再吻到你')])
+    OrderedDict({60000, '再等直至再吻到你', 70000, '再等直至再吻到你', 90000, '再等直至再吻到你'})
     """
     def to_mileseconds(time_str):
         mileseconds = 0
