diff -ru deepin-music-7.0.3.orig/src/libdmusic/core/audioanalysis.cpp deepin-music-7.0.3/src/libdmusic/core/audioanalysis.cpp
--- deepin-music-7.0.3.orig/src/libdmusic/core/audioanalysis.cpp	2024-01-24 20:29:50.056101703 +0100
+++ deepin-music-7.0.3/src/libdmusic/core/audioanalysis.cpp	2024-01-24 20:30:18.619513350 +0100
@@ -174,9 +174,9 @@
     meta.length = t_audioProperties->length() * 1000;
 
     bool encode = true;
-    encode &= tag->title().isNull() ? true : tag->title().isLatin1();
-    encode &= tag->artist().isNull() ? true : tag->artist().isLatin1();
-    encode &= tag->album().isNull() ? true : tag->album().isLatin1();
+    encode &= tag->title().isEmpty() ? true : tag->title().isLatin1();
+    encode &= tag->artist().isEmpty() ? true : tag->artist().isLatin1();
+    encode &= tag->album().isEmpty() ? true : tag->album().isLatin1();
     if (encode) {
         if (detectCodec.isEmpty()) {
             detectByte += tag->title().toCString();
