Index: nodeenv-1.7.0/nodeenv.py
===================================================================
--- nodeenv-1.7.0.orig/nodeenv.py
+++ nodeenv-1.7.0/nodeenv.py
@@ -545,6 +545,7 @@ def get_node_bin_url(version):
         'arm64/v8': 'arm64',
         'armv8': 'arm64',
         'armv8.4': 'arm64',
+        'loongarch64': 'loong64',
         'ppc64le': 'ppc64le',   # Power PC
         's390x': 's390x',       # IBM S390x
     }
