diff --git a/.kde-ci.yml b/.kde-ci.yml
index cc8b205..a617a69 100644
--- a/.kde-ci.yml
+++ b/.kde-ci.yml
@@ -9,7 +9,6 @@ Dependencies:
       'frameworks/kirigami': '@latest-kf6'
       'frameworks/kcoreaddons': '@latest-kf6'
       'frameworks/kconfig': '@latest-kf6'
-      'frameworks/knotifications': '@latest-kf6'
       'libraries/kirigami-addons': '@latest-kf6'
 
   - 'on': [ 'Linux/Qt6', 'FreeBSD/Qt6', 'Windows/Qt6' ]
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d726721..7cf4a4f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ ecm_setup_version(${PROJECT_VERSION}
 )
 
 find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Quick Gui Qml QuickControls2 Svg)
-find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Kirigami CoreAddons Config I18n Notifications)
+find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Kirigami CoreAddons Config I18n)
 
 if (BUILD_TESTING)
     find_package(Qt6Test ${QT_MIN_VERSION} REQUIRED)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8a67ec0..bfb8544 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -18,7 +18,6 @@ target_link_libraries(${CMAKE_PROJECT_NAME}_static PUBLIC
     KF6::CoreAddons
     KF6::ConfigCore
     KF6::ConfigGui
-    KF6::Notifications
 )
 
 add_executable(${CMAKE_PROJECT_NAME}
