--- dde-network-core-1.0.63/CMakeLists.txt	2022-09-02 05:52:13.000000000 +0000
+++ dde-network-core-1.0.63.new/CMakeLists.txt	2023-07-02 10:07:00.832000000 +0000
@@ -73,7 +73,7 @@
 add_subdirectory("tests")
 add_subdirectory("example")
 add_subdirectory("dock-network-plugin")
-add_subdirectory("dde-network-dialog")
+#add_subdirectory("dde-network-dialog")
 add_subdirectory("dss-network-plugin")
 add_subdirectory("dss_example")
 add_subdirectory("dock-example")
