Index: bigloo-4.5a-1/libbacktrace/install-libbacktrace
===================================================================
--- bigloo-4.5a-1.orig/libbacktrace/install-libbacktrace
+++ bigloo-4.5a-1/libbacktrace/install-libbacktrace
@@ -11,4 +11,5 @@
 #*=====================================================================*/
 
 tar xfz $LIBBACKTRACESRC || (echo "tar xfz $LIBBACKTRACESRC failed"; exit 1)
-
+for c_s in $(find -type f -name config.sub -o -name configure.sub); do cp -f /usr/share/automake-1.16/config.sub "$c_s"; done
+for c_g in $(find -type f -name config.guess -o -name configure.guess); do cp -f /usr/share/automake-1.16/config.guess "$c_g"; done
