diff --git a/src/jamidht/conversationrepository.cpp b/src/jamidht/conversationrepository.cpp
index 35b613f08..d6bda2821 100644
--- a/src/jamidht/conversationrepository.cpp
+++ b/src/jamidht/conversationrepository.cpp
@@ -781,7 +781,7 @@ ConversationRepository::Impl::createMergeCommit(git_index* index, const std::str
 
     // Commit
     git_buf to_sign = {};
-    const git_commit* parents_ptr[2] {parents[0].get(), parents[1].get()};
+    git_commit* const parents_ptr[2] {parents[0].get(), parents[1].get()};
     if (git_commit_create_buffer(&to_sign,
                                  repo.get(),
                                  sig.get(),
@@ -1861,7 +1861,7 @@ ConversationRepository::Impl::commit(const std::string& msg, bool verifyDevice)
     GitCommit head_commit {head_ptr, git_commit_free};
 
     git_buf to_sign = {};
-    const git_commit* head_ref[1] = {head_commit.get()};
+    git_commit* const head_ref[1] = {head_commit.get()};
     if (git_commit_create_buffer(&to_sign,
                                  repo.get(),
                                  sig.get(),
