--- webkitgtk-2.42.3/Source/cmake/FindGStreamer.cmake	2023-12-27 00:33:15.109385432 +0800
+++ webkitgtk-2.42.3/Source/cmake/FindGStreamer.cmake	2023-12-27 00:33:15.129385924 +0800
@@ -65,7 +65,7 @@
 
     string(REGEX MATCH "(.*)>=(.*)" _dummy "${_pkgconfig_name}")
     if ("${CMAKE_MATCH_2}" STREQUAL "")
-        pkg_check_modules(PC_${_component_prefix} "${_pkgconfig_name} >= ${GStreamer_FIND_VERSION}")
+        pkg_check_modules(PC_${_component_prefix} "${_pkgconfig_name}>=${GStreamer_FIND_VERSION}")
     else ()
         pkg_check_modules(PC_${_component_prefix} ${_pkgconfig_name})
     endif ()
