--- a/src/engine/sample.cpp
+++ b/src/engine/sample.cpp
@@ -207,7 +207,7 @@ DivDataErrors DivSample::readSampleData(SafeReader& reader, short version) {

     // render data
     if (depth!=DIV_SAMPLE_DEPTH_8BIT && depth!=DIV_SAMPLE_DEPTH_16BIT) {
-      logW("sample depth is wrong! (%d)",depth);
+      logW("sample depth is wrong! (%d)",(int)depth);
       depth=DIV_SAMPLE_DEPTH_16BIT;
     }
     samples=(double)samples/samplePitchesSD[pitch];
--- a/src/engine/zsmOps.cpp
+++ b/src/engine/zsmOps.cpp
@@ -52,7 +52,7 @@ SafeWriter* DivEngine::saveZSM(unsigned int zsmrate, bool loop) {
         break;
       default:
         IGNORED++;
-        logD("Ignoring chip %d systemID %d",i,song.system[i]);
+        logD("Ignoring chip %d systemID %d",i,(int)song.system[i]);
         break;
     }
   }
