--- liborcus-0.17.2/include/orcus/types.hpp	2023-03-10 19:40:51.890866647 +0800
+++ liborcus-0.17.2/include/orcus/types.hpp	2023-03-10 19:41:05.074252753 +0800
@@ -12,6 +12,7 @@
 #include <vector>
 #include <string>
 #include <unordered_set>
+#include <cstdint>
 #include "env.hpp"
 
 namespace orcus {
--- liborcus-0.17.2/include/orcus/base64.hpp	2021-10-14 10:59:58.000000000 +0800
+++ liborcus-0.17.2/include/orcus/base64.hpp	2023-03-10 19:47:14.812579067 +0800
@@ -11,6 +11,7 @@
 #include "env.hpp"
 #include <vector>
 #include <string>
+#include <cstdint>
 
 namespace orcus {
 
