--- compute-runtime-23.13.26032.30/third_party/DirectX-Headers/include/wsl/winadapter.h.bak	2023-06-02 08:31:26.316546560 +0200
+++ compute-runtime-23.13.26032.30/third_party/DirectX-Headers/include/wsl/winadapter.h	2023-06-02 08:31:50.872805412 +0200
@@ -75,6 +75,9 @@
 #define DECLSPEC_NOVTABLE
 #define DECLSPEC_SELECTANY
 #define EXTERN_C extern "C"
+#ifdef APIENTRY
+#undef APIENTRY
+#endif
 #define APIENTRY
 #define OUT
 #define IN
