From 1a0562ba2f8e3a1b56c027a71514bbb30e3cecc1 Mon Sep 17 00:00:00 2001
From: asciiwolf <mail@asciiwolf.com>
Date: Fri, 13 Oct 2023 15:04:09 +0000
Subject: [PATCH] Fix TBB archive name format

---
 torbrowser_launcher/common.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index 6623c0cd..9d2fef4f 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -104,11 +104,11 @@ def build_paths(self, tbb_version=None):
         if tbb_version:
             # tarball filename
             if self.architecture == "x86_64":
-                arch = "linux64"
+                arch = "linux-x86_64"
             else:
-                arch = "linux32"
+                arch = "linux-i686"
 
-            tarball_filename = "tor-browser-" + arch + "-" + tbb_version + "_ALL.tar.xz"
+            tarball_filename = "tor-browser-" + arch + "-" + tbb_version + ".tar.xz"
 
             # tarball
             self.paths["tarball_url"] = (
