diff --git a/src/sfizz/Synth.cpp b/src/sfizz/Synth.cpp
index a484cb0c..b3b796cb 100644
--- a/src/sfizz/Synth.cpp
+++ b/src/sfizz/Synth.cpp
@@ -34,6 +34,7 @@
 #include <absl/algorithm/container.h>
 #include <absl/memory/memory.h>
 #include <absl/strings/str_replace.h>
+#include <absl/strings/str_cat.h>
 #include <absl/types/optional.h>
 #include <absl/types/span.h>
 #include <algorithm>
diff --git a/src/sfizz/modulations/ModMatrix.cpp b/src/sfizz/modulations/ModMatrix.cpp
index 98a92c86..bb1e842f 100644
--- a/src/sfizz/modulations/ModMatrix.cpp
+++ b/src/sfizz/modulations/ModMatrix.cpp
@@ -14,6 +14,7 @@
 #include "utility/Debug.h"
 #include <absl/container/flat_hash_map.h>
 #include <absl/strings/string_view.h>
+#include <absl/strings/str_cat.h>
 #include <vector>
 #include <algorithm>
 
diff --git a/tests/ParsingT.cpp b/tests/ParsingT.cpp
index 36f01dd4..6e387764 100644
--- a/tests/ParsingT.cpp
+++ b/tests/ParsingT.cpp
@@ -10,6 +10,7 @@
 #include <iostream>
 #include "catch2/catch.hpp"
 #include "absl/strings/string_view.h"
+#include "absl/strings/str_cat.h"
 using namespace Catch::literals;
 
 struct ParsingMocker: sfz::Parser::Listener
