diff --git a/data/meson.build b/data/meson.build
index 2e4b5a7..aa96589 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -20,7 +20,7 @@ desktop_in = configure_file(input: 'gnome-nettool.desktop.in.in',
                             output: 'gnome-nettool.desktop.in',
                             configuration: desktop_conf)
 
-i18n.merge_file('desktop',
+i18n.merge_file(
                 type: 'desktop',
                 input: desktop_in,
                 output: 'gnome-nettool.desktop',
@@ -28,7 +28,7 @@ i18n.merge_file('desktop',
                 install: true,
                 install_dir: join_paths(prefix, datadir, 'applications'))
 
-i18n.merge_file('desktop',
+i18n.merge_file(
                 type: 'xml',
                 input: 'gnome-nettool.appdata.xml.in',
                 output: 'gnome-nettool.appdata.xml',
