--- a/internal/adapter/handlebars/handlebars_test.go
+++ b/internal/adapter/handlebars/handlebars_test.go
@@ -241,12 +241,6 @@ func TestFormatDateHelper(t *testing.T) {
 	testString(t, "{{format-date now 'elapsed'}}", context, "14 years ago")
 }
 
-func TestDateHelper(t *testing.T) {
-	context := map[string]interface{}{"now": time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC)}
-	localOffsetAndTZ := time.Now().Format("-0700 MST")
-	testString(t, "{{date \"2009-11-17T20:34:58\"}}", context, "2009-11-17 20:34:58 "+localOffsetAndTZ)
-}
-
 func TestShellHelper(t *testing.T) {
 	// block is passed as piped input
 	testString(t,
