From 4989dde69637d6172a6c866df1c33a9e5a04902f Mon Sep 17 00:00:00 2001
From: Serge Lu <Serge.Lu@amd.com>
Date: Wed, 6 Sep 2023 03:15:58 +0800
Subject: [PATCH] Use default_factory of dataclasses for mutables

---
 tensilelite/Tensile/KernelWriter.py | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/tensilelite/Tensile/KernelWriter.py b/tensilelite/Tensile/KernelWriter.py
index d165eedcd..5b3278ee9 100644
--- a/tensilelite/Tensile/KernelWriter.py
+++ b/tensilelite/Tensile/KernelWriter.py
@@ -186,13 +186,13 @@ class StateValues:
   useAtomicAdd: bool                     = False
   serializedStore: bool                  = False
 
-  a: ABMatrixInfo = ABMatrixInfo()
-  b: ABMatrixInfo = ABMatrixInfo()
-  c: MatrixInfo = MatrixInfo()
-  d: MatrixInfo = MatrixInfo()
-  e: MatrixInfo = MatrixInfo()
-  bias: MatrixInfo = MatrixInfo()
-  m: ABMatrixInfo = ABMatrixInfo()       # For Sparse Metadata
+  a: ABMatrixInfo                        = field(default_factory=ABMatrixInfo)
+  b: ABMatrixInfo                        = field(default_factory=ABMatrixInfo)
+  c: MatrixInfo                          = field(default_factory=MatrixInfo)
+  d: MatrixInfo                          = field(default_factory=MatrixInfo)
+  e: MatrixInfo                          = field(default_factory=MatrixInfo)
+  bias: MatrixInfo                       = field(default_factory=MatrixInfo)
+  m: ABMatrixInfo                        = field(default_factory=ABMatrixInfo)       # For Sparse Metadata
   totalAgprs: int                        = 0
   totalVgprs: int                        = 0
   totalSgprs: int                        = 0
