Index: zabbix-6.2.1/src/go/plugins/system/uname/uname_int8.go
===================================================================
--- zabbix-6.2.1.orig/src/go/plugins/system/uname/uname_int8.go
+++ zabbix-6.2.1/src/go/plugins/system/uname/uname_int8.go
@@ -1,5 +1,5 @@
-//go:build (linux && 386) || (linux && amd64) || (linux && arm64) || (linux && mips64le) || (linux && mipsle)
-// +build linux,386 linux,amd64 linux,arm64 linux,mips64le linux,mipsle
+//go:build (linux && 386) || (linux && amd64) || (linux && arm64) || (linux && mips64le) || (linux && mipsle) || (linux && loong64)
+// +build linux,386 linux,amd64 linux,arm64 linux,mips64le linux,mipsle linux,loong64
 
 /*
 ** Zabbix
