From ab067e45460d9b8619344b814f2fe21f0a5d5cb9 Mon Sep 17 00:00:00 2001
From: H3o66 <chris.h3o66@gmail.com>
Date: Mon, 6 May 2024 21:37:14 +0200
Subject: [PATCH] fix: add algorithm include to fix gcc14 compile

---
 src/openrct2/core/FileWatcher.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/openrct2/core/FileWatcher.cpp b/src/openrct2/core/FileWatcher.cpp
index afb9177c1818..b3c24db6c65f 100644
--- a/src/openrct2/core/FileWatcher.cpp
+++ b/src/openrct2/core/FileWatcher.cpp
@@ -7,6 +7,7 @@
  * OpenRCT2 is licensed under the GNU General Public License version 3.
  *****************************************************************************/
 
+#include <algorithm>
 #include <array>
 #include <cstdio>
 #include <stdexcept>
