diff --git a/src/python/distutils-setup.py b/src/python/distutils-setup.py
index 01ed6b58d..469b35956 100644
--- a/src/python/distutils-setup.py
+++ b/src/python/distutils-setup.py
@@ -10,7 +10,7 @@
 # under the terms of the GNU General Public License as published by the
 # Free Software Foundation; either version 2 of the License, or (at your
 # option) any later version.
-# 
+#
 # This program is distributed in the hope that it will be useful, but
 # WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 # or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
@@ -18,7 +18,7 @@
 #
 
 # Old way, only used by platforms without setuptools module
-from distutils.core import setup, Extension
+from setuptools.core import setup, Extension
 
 setup(name = 'pcp',
     version = '5.0',
