Index: tracker-miners/docs/manpages/meson.build
===================================================================
--- tracker-miners.orig/docs/manpages/meson.build
+++ tracker-miners/docs/manpages/meson.build
@@ -43,7 +43,7 @@ foreach m : daemon_manpages + cli_manpag
     command: [xsltproc,
               '--output', '@OUTPUT@',
               '--stringparam', 'man.authors.section.enabled', '0',
-              '/etc/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
+              '/usr/lib/python3.10/site-packages/asciidoc/resources/docbook-xsl/manpage.xsl', '@INPUT@'],
     input: xml,
     output: manpage,
     install: true,
