diff --git a/data/appdata/meson.build b/data/appdata/meson.build
index 1f5744e..d2c2c3a 100644
--- a/data/appdata/meson.build
+++ b/data/appdata/meson.build
@@ -3,7 +3,6 @@ metainfo_conf.set('app-id', application_id)
 metainfo_conf.set('gettext-package', gettext_package)
 
 metainfo_file = i18n.merge_file(
-  'metainfo-file',
   input: configure_file(
   	input: 'org.gnome.SoundRecorder.metainfo.xml.in.in',
   	output: '@BASENAME@',
diff --git a/data/meson.build b/data/meson.build
index 44117ad..0ff8fa5 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -14,7 +14,6 @@ install_data(
 desktop_conf = configuration_data()
 desktop_conf.set('app-id', application_id)
 desktop_file = i18n.merge_file(
-  'desktop',
   input: configure_file(
   	input: 'org.gnome.SoundRecorder.desktop.in.in',
   	output: '@BASENAME@',
