Index: jdk19u-jdk-19.0.1-10/src/hotspot/os/linux/os_linux.cpp
===================================================================
--- jdk19u-jdk-19.0.1-10.orig/src/hotspot/os/linux/os_linux.cpp
+++ jdk19u-jdk-19.0.1-10/src/hotspot/os/linux/os_linux.cpp
@@ -1670,11 +1670,11 @@ void * os::dll_load(const char *filename
   static  Elf32_Half running_arch_code=EM_SH;
 #elif  (defined RISCV)
   static  Elf32_Half running_arch_code=EM_RISCV;
-#elif  (defined LOONGARCH)
+#elif  (defined LOONGARCH64)
   static  Elf32_Half running_arch_code=EM_LOONGARCH;
 #else
     #error Method os::dll_load requires that one of following is defined:\
-        AARCH64, ALPHA, ARM, AMD64, IA32, IA64, LOONGARCH, M68K, MIPS, MIPSEL, PARISC, __powerpc__, __powerpc64__, RISCV, S390, SH, __sparc
+        AARCH64, ALPHA, ARM, AMD64, IA32, IA64, LOONGARCH64, M68K, MIPS, MIPSEL, PARISC, __powerpc__, __powerpc64__, RISCV, S390, SH, __sparc
 #endif
 
   // Identify compatibility class for VM's architecture and library's architecture
