--- FreeRDP-3.5.1/winpr/libwinpr/utils/image.c	2024-08-27 11:13:34.908362095 +0000
+++ FreeRDP-3.5.1/winpr/libwinpr/utils/image.c	2024-08-27 11:14:10.912456969 +0000
@@ -627,7 +627,7 @@
 	{
 		const JDIMENSION offset = x * stride;
 		const JSAMPLE* coffset = &cdata[offset];
-		if (jpeg_write_scanlines(&cinfo, &coffset, 1) != 1)
+		if (jpeg_write_scanlines(&cinfo, (unsigned char **)&coffset, 1) != 1)
 			goto fail;
 	}
 
