From b843302d78ee7effa604088187a1039c181c2f6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 6 Aug 2024 10:56:18 +0200
Subject: [PATCH] Fix incompatible pointer

---
 libgiggle/giggle-plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgiggle/giggle-plugin.c b/libgiggle/giggle-plugin.c
index 213368b..0979238 100644
--- a/libgiggle/giggle-plugin.c
+++ b/libgiggle/giggle-plugin.c
@@ -517,7 +517,7 @@ giggle_plugin_new_from_file (const char  *filename,
 	object = gtk_builder_get_object (builder, "plugin");
 
 	if (object) {
-		plugin = g_object_ref (object);
+		plugin = GIGGLE_PLUGIN (g_object_ref (object));
 		giggle_plugin_set_builder (plugin, builder);
 		giggle_plugin_set_filename (plugin, filename);
 	} else {
-- 
2.45.2

