diff --git a/tests/meson.build b/tests/meson.build
index 1b2e81a..bbfc668 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -42,7 +42,7 @@ if check.found()
       c_args: defines + flags
     )
     test('session', xvfb,
-      args: ['-a', '-s', '-screen 0 1400x900x24 -ac +extension GLX +render -noreset', session],
+      args: ['-d', '-s', '-screen 0 1400x900x24 -ac +extension GLX +render -noreset', session],
       timeout: 60*60
     )
