Index: pkgstats/internal/system/system_loong64.go
===================================================================
--- /dev/null
+++ pkgstats/internal/system/system_loong64.go
@@ -0,0 +1,5 @@
+package system
+
+func (system *System) GetCpuArchitecture() (string, error) {
+	return "loong64", nil
+}
Index: pkgstats/internal/system/system_loong64_test.go
===================================================================
--- /dev/null
+++ pkgstats/internal/system/system_loong64_test.go
@@ -0,0 +1,18 @@
+package system
+
+import (
+	"testing"
+)
+
+func TestGetCpuArchitecture(t *testing.T) {
+	system := System{}
+
+	cpuArch, err := system.GetCpuArchitecture()
+
+	if err != nil {
+		t.Error(err)
+	}
+	if cpuArch != "loong64" {
+		t.Error(cpuArch)
+	}
+}
