--- gitea/Makefile	2023-02-24 16:21:03.328188228 +0800
+++ gitea/Makefile	2023-02-24 16:20:25.866272584 +0800
@@ -25,7 +25,7 @@
 
 XGO_VERSION := go-1.19.x
 
-AIR_PACKAGE ?= github.com/cosmtrek/air@v1.40.4
+AIR_PACKAGE ?= github.com/cosmtrek/air@v1.41.0
 EDITORCONFIG_CHECKER_PACKAGE ?= github.com/editorconfig-checker/editorconfig-checker/cmd/editorconfig-checker@2.5.0
 ERRCHECK_PACKAGE ?= github.com/kisielk/errcheck@v1.6.1
 GOFUMPT_PACKAGE ?= mvdan.cc/gofumpt@v0.3.1
