diff --git i/src/main.cpp w/src/main.cpp
index de93e98..cd967c2 100644
--- i/src/main.cpp
+++ w/src/main.cpp
@@ -109,18 +109,7 @@ static bool checkLocation(REAPER_PLUGIN_HINSTANCE module)
   const Path &current = FS::canonical({info.dli_fname});
 #endif
 
-  if(current == expected)
-    return true;
-
-  Win32::messageBox(Splash_GetWnd(), String::format(
-    "ReaPack was not loaded from the standard extension path"
-    " or its filename was altered.\n"
-    "Move or rename it to the expected location and retry.\n\n"
-    "Current: %s\n\nExpected: %s",
-    current.join().c_str(), expected.join().c_str()
-  ).c_str(), "ReaPack: Installation path mismatch", MB_OK);
-
-  return false;
+  return true;
 }
 
 extern "C" REAPER_PLUGIN_DLL_EXPORT int REAPER_PLUGIN_ENTRYPOINT(
