diff --git a/mimosa/rpc/CMakeLists.txt b/mimosa/rpc/CMakeLists.txt
index 40a5415..a33b5b1 100644
--- a/mimosa/rpc/CMakeLists.txt
+++ b/mimosa/rpc/CMakeLists.txt
@@ -1,3 +1,5 @@
+find_package(protobuf CONFIG)
+
 install(FILES
   basic-call.hh
   bencode.hh
@@ -22,7 +24,7 @@ add_library(mimosa-rpc STATIC
   server.cc
   service.cc
   service-map.cc)
-target_link_libraries(mimosa-rpc mimosa protobuf)
+target_link_libraries(mimosa-rpc mimosa protobuf::libprotobuf)
 
 add_library(mimosa-rpc-bencode STATIC bencode.cc)
 target_link_libraries(mimosa-rpc-bencode mimosa-rpc mimosa-bencode)
diff --git a/mimosa/rpc/gen/CMakeLists.txt b/mimosa/rpc/gen/CMakeLists.txt
index f8711eb..3476c78 100644
--- a/mimosa/rpc/gen/CMakeLists.txt
+++ b/mimosa/rpc/gen/CMakeLists.txt
@@ -1,5 +1,7 @@
+find_package(protobuf CONFIG)
+
 add_executable(protoc-gen-mimosa protoc-gen-mimosa.cc)
-target_link_libraries(protoc-gen-mimosa protoc protobuf pthread z)
+target_link_libraries(protoc-gen-mimosa protoc protobuf::libprotobuf pthread z)
 
 install(TARGETS protoc-gen-mimosa
   RUNTIME DESTINATION bin
diff --git a/mimosa/rpc/gen/protoc-gen-mimosa.cc b/mimosa/rpc/gen/protoc-gen-mimosa.cc
index 702a7c0..c6e3924 100644
--- a/mimosa/rpc/gen/protoc-gen-mimosa.cc
+++ b/mimosa/rpc/gen/protoc-gen-mimosa.cc
@@ -4,7 +4,7 @@
 #include <zlib.h> // For crc32
 
 #include <google/protobuf/compiler/code_generator.h>
-#include <google/protobuf/compiler/cpp/cpp_generator.h>
+#include <google/protobuf/compiler/cpp/generator.h>
 #include <google/protobuf/compiler/plugin.h>
 #include <google/protobuf/descriptor.h>
 #include <google/protobuf/descriptor.pb.h>
