--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,15 +39,15 @@ FetchContent_Declare(juce
     SUBBUILD_DIR   "${FETCHCONTENT_BASE_DIR}/JUCE-Subbuild"
     BINARY_DIR     "${FETCHCONTENT_BASE_DIR}/JUCE-Build")
 
-FetchContent_Declare(juce-clap-extensions
+FetchContent_Declare(juce_clap
     GIT_REPOSITORY https://github.com/free-audio/clap-juce-extensions.git
-    GIT_TAG        54d83e121545490fdbb2d3dec3ba0c1da6005dde
+    GIT_TAG        0a3e7a6494788ea525e95e12a8b6a51640dcdc3e
     GIT_SHALLOW    TRUE
     GIT_CONFIG     advice.detachedHead=false)    # Disable detached HEAD warning for fetching a specific tag
 
 
 FetchContent_MakeAvailable(juce)
-FetchContent_MakeAvailable(juce-clap-extensions)
+FetchContent_MakeAvailable(juce_clap)
 
 
 ### Plugin Project Setup ###
