--- a/etc/tinyproxy.conf.in
+++ b/etc/tinyproxy.conf.in
@@ -12,8 +12,8 @@
 # as the root user. Either the user or group name or the UID or GID
 # number may be used.
 #
-User nobody
-Group nobody
+User tinyproxy
+Group tinyproxy
 
 #
 # Port: Specify the port which tinyproxy will listen on.  Please note
@@ -27,7 +27,7 @@ Port 8888
 # only one. If this is commented out, tinyproxy will bind to all
 # interfaces present.
 #
-#Listen 192.168.0.1
+Listen 127.0.0.1
 
 #
 # Bind: This allows you to specify which interface will be used for
@@ -100,7 +100,7 @@ StatFile "@pkgdatadir@/stats.html"
 # option must not be enabled if the Logfile directive is being used.
 # These two directives are mutually exclusive.
 #
-#Syslog On
+Syslog On
 
 #
 # LogLevel: Warning
@@ -124,7 +124,7 @@ LogLevel Info
 # can be used for signalling purposes.
 # If not specified, no pidfile will be written.
 #
-#PidFile "@localstatedir@/run/tinyproxy/tinyproxy.pid"
+PidFile "/run/tinyproxy/tinyproxy.pid"
 
 #
 # XTinyproxy: Tell Tinyproxy to include the X-Tinyproxy header, which
