From d055f498615c9acc7288196c88a30ec7d7670e37 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 8 Aug 2024 18:35:20 +0200
Subject: [PATCH] Mark Cancel button translatable

---
 src/transmageddon.ui | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/transmageddon.ui b/src/transmageddon.ui
index b1cd00a..69ef743 100644
--- a/src/transmageddon.ui
+++ b/src/transmageddon.ui
@@ -315,7 +315,7 @@
             </child>
             <child>
               <object class="GtkButton" id="cancelbutton">
-                <property name="label">Cancel</property>
+                <property name="label" translatable="yes">Cancel</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">True</property>
-- 
2.45.2

