diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc3021f5..252ea7ee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -187,7 +187,9 @@ MESSAGE(STATUS "UPDATE TRANSLATIONS: ${UPDATE_TRANSLATIONS}")
 set(CMAKE_AUTOMOC TRUE)
 
 # Find other needed libraries
-FIND_PACKAGE(Protobuf REQUIRED)
+FIND_PACKAGE(Protobuf CONFIG REQUIRED)
+FIND_PACKAGE(Protobuf REQUIRED) # for the protoc command
+set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
 IF(NOT EXISTS "${Protobuf_PROTOC_EXECUTABLE}")
     MESSAGE(FATAL_ERROR "No protoc command found!")
 ELSE()
