diff --git a/src/dci/private/ddcifileengine.cpp b/src/dci/private/ddcifileengine.cpp
index f1de7ab..abccb85 100644
--- a/src/dci/private/ddcifileengine.cpp
+++ b/src/dci/private/ddcifileengine.cpp
@@ -531,7 +531,7 @@ void DDciFileEngine::setFileName(const QString &fullPath)
     file = getDciFile(dciFilePath, QFile::exists(dciFilePath));
 }
 
-QDateTime DDciFileEngine::fileTime(QAbstractFileEngine::FileTime time) const
+QDateTime DDciFileEngine::fileTime(QFile::FileTime time) const
 {
     return QFileInfo(dciFilePath).fileTime(static_cast<QFile::FileTime>(time));
 }
diff --git a/src/dci/private/ddcifileengine_p.h b/src/dci/private/ddcifileengine_p.h
index 9b8af44..b20c455 100644
--- a/src/dci/private/ddcifileengine_p.h
+++ b/src/dci/private/ddcifileengine_p.h
@@ -95,7 +95,7 @@ public:
 
     void setFileName(const QString &fullPath) override;
 
-    QDateTime fileTime(FileTime time) const override;
+    QDateTime fileTime(QFile::FileTime time) const override;
 
     typedef DDciFileEngineIterator Iterator;
     Iterator *beginEntryList(QDir::Filters filters, const QStringList &filterNames) override;
