Index: mplayer-38322/configure
===================================================================
--- mplayer-38322.orig/configure
+++ mplayer-38322/configure
@@ -1853,6 +1853,7 @@ if test -z "$_target" ; then
       nios2) host_arch=nios2 ;;
       vax) host_arch=vax ;;
       xtensa*) host_arch=xtensa ;;
+      loongarch64) host_arch=loongarch64 ;;
       *) host_arch=UNKNOWN ;;
   esac
   }
@@ -2163,8 +2164,8 @@ def_fast_unaligned='#define HAVE_FAST_UN
 def_av_fast_unaligned='#define AV_HAVE_FAST_UNALIGNED 0'
 def_local_aligned='#define HAVE_LOCAL_ALIGNED 0'
 def_vsx='#define HAVE_VSX 0'
-arch_all='X86 IA64 SPARC ARM AVR32 SH4 PPC ALPHA MIPS PA_RISC S390 S390X VAX BFIN XTENSA TOMI GENERIC AARCH64'
-subarch_all='X86_32 X86_64 PPC64'
+arch_all='X86 IA64 SPARC ARM AVR32 SH4 PPC ALPHA MIPS PA_RISC S390 S390X VAX BFIN XTENSA TOMI GENERIC AARCH64 LOONGARCH'
+subarch_all='X86_32 X86_64 PPC64 LOONGSON LOONGARCH32 LOONGARCH64'
 case "$host_arch" in
   i[3-9]86|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*|i586-i686)
   arch='x86'
@@ -2835,6 +2836,18 @@ EOF
     arch='arc'
     iproc='arc'
     ;;
+  loongarch64)
+    arch='loongarch'
+    subarch='loongarch64'
+    def_fast_unaligned='#define HAVE_FAST_UNALIGNED 1'
+    def_av_fast_unaligned='#define AV_HAVE_FAST_UNALIGNED 1'
+    def_local_aligned_8='#define HAVE_LOCAL_ALIGNED_8 1'
+    def_local_aligned_16='#define HAVE_LOCAL_ALIGNED_16 1'
+    def_local_aligned_32='#define HAVE_LOCAL_ALIGNED_32 1'
+    def_fast_64bit='#define HAVE_FAST_64BIT 1'
+    _march='-march=loongarch'
+    iproc='loongarch'
+    ;;
 
   *)
     echo "The architecture of your CPU ($host_arch) is not supported by this configure script"
