diff -Naur libcoverart-1.0.0.orig/CMakeLists.txt libcoverart-1.0.0/CMakeLists.txt
--- libcoverart-1.0.0.orig/CMakeLists.txt	2012-10-09 14:51:32.000000000 +0200
+++ libcoverart-1.0.0/CMakeLists.txt	2024-08-03 02:17:56.037344306 +0200
@@ -33,7 +33,7 @@
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_SOURCE_DIR}/config.h)
 
 FILE(GLOB headers ${CMAKE_CURRENT_SOURCE_DIR}/include/coverart/*.h)
-INSTALL(FILES ${headers} include/coverart/caa_c.h DESTINATION ${INCLUDE_INSTALL_DIR}/coverart)
+INSTALL(FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/include/coverart/caa_c.h DESTINATION ${INCLUDE_INSTALL_DIR}/coverart)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libcoverart.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 
 ADD_SUBDIRECTORY(src)
