diff --git a/src/base/basictypes.h b/src/base/basictypes.h
index ea87a6d..ccbd2c8 100644
--- a/src/base/basictypes.h
+++ b/src/base/basictypes.h
@@ -387,6 +387,8 @@ class AssignAttributeStartEnd {
 #   define CACHELINE_ALIGNED __attribute__((aligned(64)))
 # elif (defined(__e2k__))
 #   define CACHELINE_ALIGNED __attribute__((aligned(64)))
+# elif (defined(__loongarch64))
+#   define CACHELINE_ALIGNED __attribute__((aligned(64)))
 # else
 #   error Could not determine cache line length - unknown architecture
 # endif
