Index: neofetch-7.1.0/neofetch
===================================================================
--- neofetch-7.1.0.orig/neofetch
+++ neofetch-7.1.0/neofetch
@@ -2088,8 +2089,8 @@ get_cpu() {
 
                 *)
                     cpu="$(awk -F '\\s*: | @' \
-                            '/model name|Hardware|Processor|^cpu model|chip type|^cpu type/ {
-                            cpu=$2; if ($1 == "Hardware") exit } END { print cpu }' "$cpu_file")"
+                            '/[mM]odel [nN]ame|Hardware|Processor|^cpu model|chip type|^cpu type/ {
+                            cpu=$2; if ($1 == "Hardware" || $1 == "Model Name") exit } END { print cpu }' "$cpu_file")"
                 ;;
             esac
 
