diff --git a/cowsay b/cowsay
index 900ca46..53928df 100755
--- a/cowsay
+++ b/cowsay
@@ -6,6 +6,7 @@
 ## This file is part of cowsay.  (c) 1999-2000 Tony Monroe.
 ##
 
+use Encode;
 use Text::Tabs qw(expand);
 use Text::Wrap qw(wrap fill $columns);
 use File::Basename;
@@ -103,7 +104,7 @@ sub maxlength {
     my ($l, $m);
     $m = -1;
     for my $i (@_) {
-	$l = length $i;
+	$l = length(Encode::decode_utf8($i));
 	$m = $l if ($l > $m);
     }
     return $m;
diff --git a/install.sh b/install.sh
index 0b6f2fa..4ca8c62 100755
--- a/install.sh
+++ b/install.sh
@@ -67,7 +67,7 @@ PREFIX=${prefix:-/usr/local}
 echo Okay, time to install this puppy.
 
 echo s,%BANGPERL%,!$usethisperl,\; > install.pl
-echo s,%PREFIX%,$PREFIX,\; >> install.pl
+echo s,%PREFIX%,/usr,\; >> install.pl
 set -x
 mkdir -p $PREFIX/bin || (mkdir $PREFIX; mkdir $PREFIX/bin)
 $usethisperl -p install.pl cowsay > $PREFIX/bin/cowsay
