Index: libfido2-1.5.0/CMakeLists.txt
===================================================================
--- libfido2-1.5.0.orig/CMakeLists.txt
+++ libfido2-1.5.0/CMakeLists.txt
@@ -135,10 +135,6 @@ else()
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wbad-function-cast")
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic")
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic-errors")
-	check_c_compiler_flag("-fstack-protector-all" HAVE_STACK_PROTECTOR_ALL)
-	if(HAVE_STACK_PROTECTOR_ALL)
-		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-all")
-	endif()
 
 	add_definitions(-D_DEFAULT_SOURCE)
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
@@ -190,10 +186,7 @@ if(HAVE_ERR_H)
 endif()
 
 # unistd.h
-check_include_files(unistd.h HAVE_UNISTD_H)
-if(HAVE_UNISTD_H)
-	add_definitions(-DHAVE_UNISTD_H)
-endif()
+add_definitions(-DHAVE_UNISTD_H)
 
 # signal.h
 check_include_files(signal.h HAVE_SIGNAL_H)
