diff --git i/cmake_include/AsioStandaloneConfig.cmake w/cmake_include/AsioStandaloneConfig.cmake
index 66b4d53..2b7649a 100644
--- i/cmake_include/AsioStandaloneConfig.cmake
+++ w/cmake_include/AsioStandaloneConfig.cmake
@@ -1,6 +1 @@
 add_library(AsioStandalone::AsioStandalone IMPORTED INTERFACE)
-
-set_property(TARGET AsioStandalone::AsioStandalone APPEND PROPERTY
-  INTERFACE_INCLUDE_DIRECTORIES
-  ${CMAKE_CURRENT_LIST_DIR}/../modules/asio-standalone/asio/include
-)
diff --git i/cmake_include/CatchConfig.cmake w/cmake_include/CatchConfig.cmake
index 2470e18..0a41078 100644
--- i/cmake_include/CatchConfig.cmake
+++ w/cmake_include/CatchConfig.cmake
@@ -1,6 +1 @@
-add_library(Catch::Catch IMPORTED INTERFACE)
-
-set_property(TARGET Catch::Catch APPEND PROPERTY
-  INTERFACE_INCLUDE_DIRECTORIES
-  ${CMAKE_CURRENT_LIST_DIR}/../third_party/catch
-)
+find_package(Catch2 3 REQUIRED)
diff --git i/include/ableton/test/CatchWrapper.hpp w/include/ableton/test/CatchWrapper.hpp
index 997da91..0f1d2ed 100644
--- i/include/ableton/test/CatchWrapper.hpp
+++ w/include/ableton/test/CatchWrapper.hpp
@@ -35,7 +35,7 @@
 #pragma warning(disable : 5220)
 #endif
 
-#include <catch.hpp>
+#include <catch2/catch_all.hpp>
 
 #if defined(_MSC_VER)
 #pragma warning(pop)
diff --git i/src/CMakeLists.txt w/src/CMakeLists.txt
index 9f42391..11f6a7f 100644
--- i/src/CMakeLists.txt
+++ w/src/CMakeLists.txt
@@ -77,7 +77,7 @@ source_group("Test Utils" FILES
 #                |___/
 
 function(configure_link_test_executable target)
-  target_link_libraries(${target} Catch::Catch Ableton::Link)
+  target_link_libraries(${target} Catch2::Catch2WithMain Ableton::Link)
   target_compile_definitions(${target} PRIVATE -DCATCH_CONFIG_ENABLE_BENCHMARKING=1)
 endfunction()
 
