--- Gemfile.orig	2023-08-08 16:14:14.726449938 +0000
+++ Gemfile	2023-08-08 16:14:20.603486066 +0000
@@ -336,7 +336,7 @@ gem 'sentry-sidekiq', '~> 5.8.0'
 
 # PostgreSQL query parsing
 #
-gem 'pg_query', '~> 4.2.1'
+gem 'pg_query', '~> 4.2.3'
 
 gem 'premailer-rails', '~> 1.10.3'
 
--- Gemfile.lock.orig	2023-08-08 16:17:27.631956179 +0000
+++ Gemfile.lock	2023-08-08 16:18:10.261323891 +0000
@@ -1137,7 +1137,7 @@ GEM
     peek (1.1.0)
       railties (>= 4.0.0)
     pg (1.5.3)
-    pg_query (4.2.1)
+    pg_query (4.2.3)
       google-protobuf (>= 3.22.3)
     plist (3.6.0)
     png_quantizator (0.2.1)
@@ -1871,7 +1871,7 @@ DEPENDENCIES
   parslet (~> 1.8)
   peek (~> 1.1)
   pg (~> 1.5.3)
-  pg_query (~> 4.2.1)
+  pg_query (~> 4.2.3)
   png_quantizator (~> 0.2.1)
   premailer-rails (~> 1.10.3)
   prometheus-client-mmap (~> 0.25)
--- Gemfile.checksum.orig	2023-08-08 16:25:40.960039145 +0000
+++ Gemfile.checksum	2023-08-08 16:25:10.831446239 +0000
@@ -446,7 +446,7 @@
 {"name":"pg","version":"1.5.3","platform":"x64-mingw-ucrt","checksum":"1f2a6b2afaf0ccb8afe8b6a00131bce8151fbd6e8826b2d944288f6f2b615389"},
 {"name":"pg","version":"1.5.3","platform":"x64-mingw32","checksum":"ab7f5f3020323094a2b16f9638166b04c103e152a9079a1b8e795f4bf79765e0"},
 {"name":"pg","version":"1.5.3","platform":"x86-mingw32","checksum":"aa6ddda9887462d30a6d49d875eb9d27fca8cdb7185103b650e7351b38f15ddf"},
-{"name":"pg_query","version":"4.2.1","platform":"ruby","checksum":"b04820a9d1c0c1608e3240b7d84baabbee1b95a7302f29fdd0f00e901c604833"},
+{"name":"pg_query","version":"4.2.3","platform":"ruby","checksum":"1cc9955c7bce8e51e1abc11f1952e3d9d0f1cd4c16c58c56ec75d5aaf1cfd697"},
 {"name":"plist","version":"3.6.0","platform":"ruby","checksum":"f468bcf6b72ec6d1585ed6744eb4817c1932a5bf91895ed056e69b7f12ca10f2"},
 {"name":"png_quantizator","version":"0.2.1","platform":"ruby","checksum":"6023d4d064125c3a7e02929c95b7320ed6ac0d7341f9e8de0c9ea6576ef3106b"},
 {"name":"po_to_json","version":"1.0.1","platform":"ruby","checksum":"6a7188aa6c42a22c9718f9b39062862ef7f3d8f6a7b4177cae058c3308b56af7"},
