--- MaterialX-1.38.8/source/MaterialXView/NanoGUI/CMakeLists.txt	2023-12-29 21:43:48.000000000 +0800
+++ MaterialX-1.38.8/source/MaterialXView/NanoGUI/CMakeLists.txt	2023-12-29 21:48:51.415250915 +0800
@@ -84,8 +84,6 @@
                    (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "arm64")))
   # Apple M1 compatibility
   set(NANOGUI_NATIVE_FLAGS_DEFAULT "-mcpu=apple-a12")
-else()
-  set(NANOGUI_NATIVE_FLAGS_DEFAULT "-march=nehalem")
 endif()
 
 option(NANOGUI_BUILD_EXAMPLES            "Build NanoGUI example application?" ON)
