diff --git a/tests/integration/test_validators.py b/tests/integration/test_validators.py
index 07bc4df..5663c4c 100644
--- a/tests/integration/test_validators.py
+++ b/tests/integration/test_validators.py
@@ -863,5 +863,5 @@ class TestOAS31ValidatorValidate(BaseTestOASValidatorValidate):
         with pytest.raises(ValidationError) as excinfo:
             validator.validate(value)
 
-        error = "Expected at most 4 items, but found 5"
+        error = "Expected at most 4 items but found 1 extra"
         assert error in str(excinfo.value)
