diff --unified --recursive --text --new-file symlinks-1.4.3.orig/Makefile symlinks-1.4.3/Makefile
--- symlinks-1.4.3.orig/Makefile	2024-07-23 07:16:07.528608342 +0200
+++ symlinks-1.4.3/Makefile	2024-07-23 07:15:38.631804272 +0200
@@ -10,7 +11,7 @@
 all: symlinks
 
 symlinks: symlinks.c
-	$(CC) -Wall -Wstrict-prototypes -O2 $(CFLAGS) -o symlinks symlinks.c
+	$(CC) -Wall -Wstrict-prototypes -O2 $(CFLAGS) $(LDFLAGS) -o symlinks symlinks.c
 
 install: all symlinks.8
 	$(INSTALL) -c -o $(OWNER) -g $(GROUP) -m 755 symlinks $(BINDIR)
