diff -aur DisplayCAL-3.9.10/DisplayCAL/meta.py DisplayCAL-3.9.10.old/DisplayCAL/meta.py
--- DisplayCAL-3.9.10/DisplayCAL/meta.py	2023-06-05 19:44:28.927539723 +0200
+++ DisplayCAL-3.9.10.old/DisplayCAL/meta.py	2023-06-05 19:44:00.982887386 +0200
@@ -62,7 +62,7 @@
 name_html = '<span class="appname">Display<span>CAL</span></span>'
 
 py_minversion = (3, 8)
-py_maxversion = (3, 12)
+py_maxversion = (3, 10)
 
 version = VERSION_STRING
 version_lin = VERSION_STRING  # Linux
