diff --git a/Make.defaults b/Make.defaults
index 1277d14..6af5acf 100755
--- a/Make.defaults
+++ b/Make.defaults
@@ -197,7 +197,7 @@ endif
 ARFLAGS := rDv
 ASFLAGS += $(ARCH3264)
 LDFLAGS	+= -nostdlib --warn-common --no-undefined --fatal-warnings \
-	   --build-id=sha1
+	   --build-id=sha1 --no-warn-rwx-segments
 
 ifneq ($(ARCH),arm)
 export LIBGCC=$(shell $(CC) $(CFLAGS) $(ARCH3264) -print-libgcc-file-name)
diff --git a/inc/loongarch64/efibind.h b/inc/loongarch64/efibind.h
index aaf3fb7..65cbdfd 100644
--- a/inc/loongarch64/efibind.h
+++ b/inc/loongarch64/efibind.h
@@ -45,6 +45,9 @@ typedef int64_t             intptr_t;
 #ifndef __WCHAR_TYPE__
 # define __WCHAR_TYPE__ short
 #endif
+#ifndef __CHAR16_TYPE__
+# define __CHAR16_TYPE__ unsigned short
+#endif
 
 typedef uint64_t   UINT64;
 typedef int64_t    INT64;
@@ -53,8 +56,10 @@ typedef uint32_t   UINT32;
 typedef int32_t    INT32;
 
 typedef uint16_t   UINT16;
+typedef __CHAR16_TYPE__ CHAR16;
 typedef int16_t    INT16;
 typedef uint8_t    UINT8;
+typedef char       CHAR8;
 typedef int8_t     INT8;
 typedef __WCHAR_TYPE__ WCHAR;
 
