diff --git a/.github/workflows/loongarch64.yml b/.github/workflows/loongarch64.yml
index 4a9bf98b6..b310d6938 100644
--- a/.github/workflows/loongarch64.yml
+++ b/.github/workflows/loongarch64.yml
@@ -40,8 +40,8 @@ jobs:
 
       - name: Download and install loongarch64-toolchain
         run: |
-          wget https://github.com/loongson/build-tools/releases/download/2022.09.06/loongarch64-clfs-7.3-cross-tools-gcc-glibc.tar.xz
-          tar -xf loongarch64-clfs-7.3-cross-tools-gcc-glibc.tar.xz -C /opt
+          wget https://github.com/loongson/build-tools/releases/download/2023.08.08/CLFS-loongarch64-8.1-x86_64-cross-tools-gcc-glibc.tar.xz
+          tar -xf CLFS-loongarch64-8.1-x86_64-cross-tools-gcc-glibc.tar.xz -C /opt
 
       - name: Set env
         run: |
diff --git a/cmake/cc.cmake b/cmake/cc.cmake
index 00952e810..242b03b5f 100644
--- a/cmake/cc.cmake
+++ b/cmake/cc.cmake
@@ -36,9 +36,9 @@ if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_C_COMPILER_ID} STREQUAL "LS
 
     if (LOONGARCH64)
       if (BINARY64)
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64d")
       else ()
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp32")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=ilp32d")
       endif ()
       set(BINARY_DEFINED 1)
     endif ()
diff --git a/cmake/fc.cmake b/cmake/fc.cmake
index c496f6368..b356dfda3 100644
--- a/cmake/fc.cmake
+++ b/cmake/fc.cmake
@@ -61,9 +61,9 @@ if (${F_COMPILER} STREQUAL "GFORTRAN" OR ${F_COMPILER} STREQUAL "F95" OR CMAKE_F
     endif ()
     if (LOONGARCH64)
       if (BINARY64)
-        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64")
+        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64d")
       else ()
-        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp32")
+        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=ilp32d")
       endif ()
     endif ()
     if (RISCV64)
