Index: OpenBLAS-0.3.26/Makefile.system
===================================================================
--- OpenBLAS-0.3.26.orig/Makefile.system
+++ OpenBLAS-0.3.26/Makefile.system
@@ -948,10 +948,7 @@ BINARY_DEFINED = 1
 endif
 
 ifeq ($(ARCH), loongarch64)
-LA64_ABI=$(shell $(CC) -mabi=lp64d -c $(TOPDIR)/cpuid_loongarch64.c -o /dev/null > /dev/null 2> /dev/null && echo lp64d)
-ifneq ($(LA64_ABI), lp64d)
-LA64_ABI=lp64
-endif
+LA64_ABI=lp64d
 CCOMMON_OPT += -march=loongarch64 -mabi=$(LA64_ABI)
 FCOMMON_OPT += -march=loongarch64 -mabi=$(LA64_ABI)
 endif
Index: OpenBLAS-0.3.26/cmake/cc.cmake
===================================================================
--- OpenBLAS-0.3.26.orig/cmake/cc.cmake
+++ OpenBLAS-0.3.26/cmake/cc.cmake
@@ -36,12 +36,7 @@ if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU
 
     if (LOONGARCH64)
       if (BINARY64)
-	CHECK_CXX_COMPILER_FLAG("-mabi=lp64d" COMPILER_SUPPORT_LP64D_ABI)
-        if(COMPILER_SUPPORT_LP64D_ABI)
 	  set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64d")
-	else()
-	  set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64")
-	endif ()
       else ()
 	CHECK_CXX_COMPILER_FLAG("-mabi=ilp32d" COMPILER_SUPPORT_ILP32D_ABI)
 	if(COMPILER_SUPPORT_ILP32D_ABI)
Index: OpenBLAS-0.3.26/cmake/fc.cmake
===================================================================
--- OpenBLAS-0.3.26.orig/cmake/fc.cmake
+++ OpenBLAS-0.3.26/cmake/fc.cmake
@@ -61,12 +61,7 @@ if (${F_COMPILER} STREQUAL "GFORTRAN" OR
     endif ()
     if (LOONGARCH64)
       if (BINARY64)
-	CHECK_CXX_COMPILER_FLAG("-mabi=lp64d" COMPILER_SUPPORT_LP64D_ABI)
-        if(COMPILER_SUPPORT_LP64D_ABI)
 	  set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64d")
-	else()
-	  set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64")
-	endif ()
       else ()
 	CHECK_CXX_COMPILER_FLAG("-mabi=ilp32d" COMPILER_SUPPORT_ILP32D_ABI)
 	if(COMPILER_SUPPORT_ILP32D_ABI)
