diff --git a/pyproject.toml b/pyproject.toml
index 030be48..7a5a6da 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,8 +13,6 @@ addopts = "--cov=httpx_ws/ --cov-report=term-missing"
 allow-direct-references = true
 
 [tool.hatch.version]
-source = "regex_commit"
-commit_extra_args = ["-e"]
 path = "httpx_ws/__init__.py"
 
 [tool.hatch.envs.default]
@@ -51,7 +49,7 @@ docs-build = "mkdocs build"
 support-legacy = true  # Create setup.py
 
 [build-system]
-requires = ["hatchling", "hatch-regex-commit"]
+requires = ["hatchling"]
 build-backend = "hatchling.build"
 
 [project]
