diff --git a/futatabi/flags.h b/futatabi/flags.h
index 6e59d9d..b4e9774 100644
--- a/futatabi/flags.h
+++ b/futatabi/flags.h
@@ -3,6 +3,7 @@
 
 #include "defs.h"
 
+#include <cstdint>
 #include <string>
 #include <unordered_map>
 
diff --git a/shared/va_resource_pool.h b/shared/va_resource_pool.h
index e8f276f..3436a74 100644
--- a/shared/va_resource_pool.h
+++ b/shared/va_resource_pool.h
@@ -6,6 +6,7 @@
 
 #include <list>
 #include <mutex>
+#include <iostream>
 
 #define CHECK_VASTATUS(va_status, func) \
 	if (va_status != VA_STATUS_SUCCESS) { \
