diff --git a/nageru/av1_encoder.cpp b/nageru/av1_encoder.cpp
index 371eb0d..f4af4b0 100644
--- a/nageru/av1_encoder.cpp
+++ b/nageru/av1_encoder.cpp
@@ -292,8 +292,8 @@ void AV1Encoder::encode_frame(AV1Encoder::QueuedFrame qf)
 		pic.cr_stride = global_flags.width / 2;  // Likewise.
 		pic.width = global_flags.width;
 		pic.height = global_flags.height;
-		pic.origin_x = 0;
-		pic.origin_y = 0;
+		pic.org_x = 0;
+		pic.org_y = 0;
 		pic.color_fmt = EB_YUV420;
 		pic.bit_depth = global_flags.bit_depth > 8 ? EB_TEN_BIT : EB_EIGHT_BIT;
 
