From 61826637969cd6cc298c34c3861b72883e889a05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 2 Aug 2024 17:50:04 +0200
Subject: [PATCH] Ingredients Viewer: Don't set empty title

---
 src/gr-ingredients-viewer.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/gr-ingredients-viewer.c b/src/gr-ingredients-viewer.c
index 9900aee0..d7dce0d3 100644
--- a/src/gr-ingredients-viewer.c
+++ b/src/gr-ingredients-viewer.c
@@ -353,9 +353,11 @@ gr_ingredients_viewer_set_title (GrIngredientsViewer *viewer,
 {
         const char *display_title;
 
-        display_title = g_dgettext (GETTEXT_PACKAGE "-data", title);
-        gtk_label_set_label (GTK_LABEL (viewer->title_label), display_title);
-        gtk_entry_set_text (GTK_ENTRY (viewer->title_entry), display_title);
+        if (title[0] != 0) {
+                display_title = g_dgettext (GETTEXT_PACKAGE "-data", title);
+                gtk_label_set_label (GTK_LABEL (viewer->title_label), display_title);
+                gtk_entry_set_text (GTK_ENTRY (viewer->title_entry), display_title);
+        }
 
         g_free (viewer->title);
         viewer->title = g_strdup (title);
-- 
GitLab

