From eac2bdc38df76bd022a5cb007d568780357c8407 Mon Sep 17 00:00:00 2001
From: Shi Pujin <shipujin@loongson.cn>
Date: Mon, 6 Feb 2023 20:24:22 +0800
Subject: [PATCH 2/2] disable-pointer-compression

---
 common.gypi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/common.gypi b/common.gypi
index d713ad5586..5540425992 100644
--- a/common.gypi
+++ b/common.gypi
@@ -112,6 +112,12 @@
         'v8_enable_pointer_compression': 0,
         'v8_enable_31bit_smis_on_64bit_arch': 0,
       }],
+      # V8 pointer compression not supports loong64 and mips64el.
+      ['target_arch in "loong64 mips64el"', {
+        'v8_enable_pointer_compression': 0,
+        'v8_enable_31bit_smis_on_64bit_arch': 0,
+        'v8_enable_sandbox': 0,
+      }],
       ['target_arch in "ppc64 s390x"', {
         'v8_enable_backtrace': 1,
       }],
-- 
2.40.0

