Index: gtk/gtk/meson.build
===================================================================
--- gtk.orig/gtk/meson.build
+++ gtk/gtk/meson.build
@@ -943,6 +943,7 @@ if not meson.is_cross_build() and build_
     command : [objcopy,
                  '--strip-all',
                  '--add-symbol','_gtk_resource_data=.data:0',
+                 '--alt-elf-eflags=0x03',
                  '@INPUT@',
                  '@OUTPUT@'])
 
Index: gtk/demos/gtk-demo/meson.build
===================================================================
--- gtk.orig/demos/gtk-demo/meson.build
+++ gtk/demos/gtk-demo/meson.build
@@ -211,6 +211,7 @@ if build_machine.system() == 'linux' and
     command : [objcopy,
                  '--strip-all',
                  '--add-symbol','_g_binary_gtkdemo_resource_data=.data:0',
+                 '--alt-elf-eflags=0x03',
                  '@INPUT@',
                  '@OUTPUT@'])
 
Index: gtk/demos/widget-factory/meson.build
===================================================================
--- gtk.orig/demos/widget-factory/meson.build
+++ gtk/demos/widget-factory/meson.build
@@ -59,6 +59,7 @@ if build_machine.system() == 'linux' and
     command : [objcopy,
                  '--strip-all',
                  '--add-symbol','_g_binary_widgetfactory_resource_data=.data:0',
+                 '--alt-elf-eflags=0x03',
                  '@INPUT@',
                  '@OUTPUT@'])
 
