diff --git a/awscli/customizations/cloudformation/yamlhelper.py b/awscli/customizations/cloudformation/yamlhelper.py
index abdc749..c260142 100644
--- a/awscli/customizations/cloudformation/yamlhelper.py
+++ b/awscli/customizations/cloudformation/yamlhelper.py
@@ -112,7 +112,7 @@ def yaml_parse(yamlstr):
     except ValueError:
         yaml = ruamel.yaml.YAML(typ="safe", pure=True)
         yaml.Constructor.add_constructor(
-            ruamel.yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+            str(ruamel.yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG),
             _dict_constructor)
         yaml.Constructor.add_multi_constructor(
             "!", intrinsics_multi_constructor)
diff --git a/awscli/customizations/eks/ordered_yaml.py b/awscli/customizations/eks/ordered_yaml.py
index 23834e0..486a7bb 100644
--- a/awscli/customizations/eks/ordered_yaml.py
+++ b/awscli/customizations/eks/ordered_yaml.py
@@ -28,7 +28,7 @@ def ordered_yaml_load(stream):
     """ Load an OrderedDict object from a yaml stream."""
     yaml = ruamel.yaml.YAML(typ="safe", pure=True)
     yaml.Constructor.add_constructor(
-        ruamel.yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+        str(ruamel.yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG),
         _ordered_constructor)
 
     return yaml.load(stream)
