Index: chmlib-0.40/src/chm_http.c
===================================================================
--- chmlib-0.40.orig/src/chm_http.c
+++ chmlib-0.40/src/chm_http.c
@@ -42,6 +42,7 @@
 /* includes for networking */
 #include <sys/socket.h>
 #include <sys/types.h>
+#include <unistd.h>
 #include <netinet/in.h>
 
 /* threading includes */
Index: chmlib-0.40/src/chm_lib.c
===================================================================
--- chmlib-0.40.orig/src/chm_lib.c
+++ chmlib-0.40/src/chm_lib.c
@@ -164,7 +164,7 @@ typedef unsigned long long      UInt64;
 
 /* x86-64 */
 /* Note that these may be appropriate for other 64-bit machines. */
-#elif __x86_64__ || __ia64__
+#elif __x86_64__ || __ia64__ || __loongarch_lp64
 typedef unsigned char           UChar;
 typedef short                   Int16;
 typedef unsigned short          UInt16;
