Index: abseil-cpp-20211102.0/absl/debugging/internal/examine_stack.cc
===================================================================
--- abseil-cpp-20211102.0.orig/absl/debugging/internal/examine_stack.cc
+++ abseil-cpp-20211102.0/absl/debugging/internal/examine_stack.cc
@@ -57,6 +57,8 @@ void* GetProgramCounter(void* vuc) {
       return reinterpret_cast<void*>(context->uc_mcontext.gregs[14]);
 #elif defined(__ia64__)
     return reinterpret_cast<void*>(context->uc_mcontext.sc_ip);
+#elif defined(__loongarch64)
+    return reinterpret_cast<void*>(context->uc_mcontext.__pc);
 #elif defined(__m68k__)
     return reinterpret_cast<void*>(context->uc_mcontext.gregs[16]);
 #elif defined(__mips__)
