diff --git a/ci/official/requirements_updater/requirements.in b/ci/official/requirements_updater/requirements.in
index 5c5892b2..0ee2eea0 100644
--- a/ci/official/requirements_updater/requirements.in
+++ b/ci/official/requirements_updater/requirements.in
@@ -1,6 +1,6 @@
 # TODO(b/304751256): Adjust the numpy pin to a single version, when ready
 numpy ~= 1.23.5 ; python_version <= "3.11"
-numpy ~= 1.26.0 ; python_version >= "3.12"
+numpy ~= 2.0.0 ; python_version >= "3.12"
 wheel ~= 0.41.2
 h5py >= 3.10.0
 lit ~= 17.0.2
@@ -25,7 +25,8 @@ tensorboard ~= 2.17.0
 # Test dependencies
 grpcio >= 1.24.3, < 2.0
 portpicker == 1.6.0
-scipy == 1.11.3
+scipy == 1.11.3 ; python_version <= "3.11"
+scipy == 1.14.0 ; python_version >= "3.12"
 requests == 2.31.0
 packaging==23.2
 setuptools==68.2.2
diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index 88da8b54..e4041765 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -86,7 +86,7 @@ REQUIRED_PACKAGES = [
     'ml_dtypes >= 0.3.1, < 0.5.0',
     # TODO(b/304751256): Adjust the numpy pin to a single version, when ready
     'numpy >= 1.23.5, < 2.0.0 ; python_version <= "3.11"',
-    'numpy >= 1.26.0, < 2.0.0 ; python_version >= "3.12"',
+    'numpy ~= 2.0.0 ; python_version >= "3.12"',
     'opt_einsum >= 2.3.2',
     'packaging',
     # pylint:disable=line-too-long
