--- aaa/setup.py	2023-11-03 11:00:28.748584492 +0800
+++ /tmp/setup.py	2023-11-03 10:56:36.364171431 +0800
@@ -193,12 +193,6 @@
 
 # ad-hoc for easy_install
 DATA_FILES = []
-if ('bdist_egg' not in sys.argv and 'bdist_wheel' not in sys.argv
-        and 'editable_wheel' not in sys.argv):
-    # generate and install brz.1 only with plain install, not the
-    # easy_install one
-    build.sub_commands.append(('build_man', lambda _: True))
-    DATA_FILES = [('man/man1', ['brz.1', 'breezy/git/git-remote-bzr.1'])]
 
 import site
 
