diff --git a/fitz/helper-select.i b/fitz/helper-select.i
index 44c8489..7125732 100644
--- a/fitz/helper-select.i
+++ b/fitz/helper-select.i
@@ -369,9 +369,9 @@ void remove_dest_range(fz_context *ctx, pdf_document *pdf, PyObject *numbers)
                     pno = pdf_lookup_page_number(ctx, pdf, target);
                 }
                 else if (pdf_is_string(ctx, dest)) {
-                    pno = pdf_lookup_anchor(ctx, pdf,
+                    pno = fz_resolve_link(ctx, &(pdf->super),
                                             pdf_to_text_string(ctx, dest),
-                                            NULL, NULL);
+                                            NULL, NULL).page;
                 }
                 if (pno < 0) { // page number lookup did not work
                     continue;
