diff --git i/fitz/helper-python.i w/fitz/helper-python.i
index 48364a8..15efe84 100644
--- i/fitz/helper-python.i
+++ w/fitz/helper-python.i
@@ -2081,7 +2081,7 @@ def get_tessdata() -> str:
     if sys.platform == "win32":
         tessdata = "C:\\Program Files\\Tesseract-OCR\\tessdata"
     else:
-        tessdata = "/usr/share/tesseract-ocr/4.00/tessdata"
+        tessdata = "/usr/share/tessdata"
 
     if os.path.exists(tessdata):
         return tessdata
diff --git i/src/__init__.py w/src/__init__.py
index 7a4adb5..3f6e433 100644
--- i/src/__init__.py
+++ w/src/__init__.py
@@ -17499,7 +17499,7 @@ def get_tessdata() -> str:
     if sys.platform == "win32":
         tessdata = "C:\\Program Files\\Tesseract-OCR\\tessdata"
     else:
-        tessdata = "/usr/share/tesseract-ocr/4.00/tessdata"
+        tessdata = "/usr/share/tessdata"
 
     if os.path.exists(tessdata):
         return tessdata
