diff --git a/setup.py b/setup.py
index 364b515..030f45e 100755
--- a/setup.py
+++ b/setup.py
@@ -1247,14 +1247,10 @@ else:
         elif darwin and platform_release_tuple() < (18,):
             # There are still of ptoblems when building on old macos.
             ret.append('libclang==14.0.6')
-        else:
-            ret.append('libclang')
         if msys2:
             print(f'msys2: pip install of swig does not build; assuming `pacman -S swig`.')
         elif openbsd:
             print(f'OpenBSD: pip install of swig does not build; assuming `pkg_add swig`.')
-        else:
-            ret.append( 'swig')
         return ret
 
 build_wheel = p.build_wheel
