diff --git a/builder/builder.ml b/builder/builder.ml
index 15ece4e6..bd44556b 100644
--- a/builder/builder.ml
+++ b/builder/builder.ml
@@ -815,8 +815,8 @@ let main () =
   delete_output_file := false;
 
   (* Print the stats calculated above. *)
-  Pervasives.flush Pervasives.stdout;
-  Pervasives.flush Pervasives.stderr;
+  Stdlib.flush Stdlib.stdout;
+  Stdlib.flush Stdlib.stderr;
 
   Option.iter print_string stats
 
diff --git a/builder/index_parser.ml b/builder/index_parser.ml
index 1c5bc26c..3e19e489 100644
--- a/builder/index_parser.ml
+++ b/builder/index_parser.ml
@@ -246,7 +246,7 @@ let get_index ~downloader ~sigchecker ?(template = false)
 
     if verbose () then (
       printf "index file (%s) after parsing (C parser):\n" uri;
-      List.iter (Index.print_entry Pervasives.stdout) entries
+      List.iter (Index.print_entry Stdlib.stdout) entries
     );
 
     entries
diff --git a/builder/simplestreams_parser.ml b/builder/simplestreams_parser.ml
index 70b44827..fe0afde5 100644
--- a/builder/simplestreams_parser.ml
+++ b/builder/simplestreams_parser.ml
@@ -149,6 +149,6 @@ let get_index ~downloader ~sigchecker { Sources.uri; proxy } =
   let entries = List.flatten (List.map scan_product_list downloads) in
   if verbose () then (
     printf "simplestreams tree (%s) after parsing:\n" uri;
-    List.iter (Index.print_entry Pervasives.stdout) entries
+    List.iter (Index.print_entry Stdlib.stdout) entries
   );
   entries
diff --git a/m4/guestfs-ocaml.m4 b/m4/guestfs-ocaml.m4
index d2bf4004..a46f4f6d 100644
--- a/m4/guestfs-ocaml.m4
+++ b/m4/guestfs-ocaml.m4
@@ -38,10 +38,10 @@ AC_ARG_ENABLE([ocaml],
     [],
     [enable_ocaml=yes])
 
-dnl OCaml >= 4.04 is required.
-ocaml_ver_str=4.04
+dnl OCaml >= 4.07 is required.
+ocaml_ver_str=4.07
 ocaml_min_major=4
-ocaml_min_minor=4
+ocaml_min_minor=7
 AC_MSG_CHECKING([if OCaml version >= $ocaml_ver_str])
 ocaml_major="`echo $OCAMLVERSION | $AWK -F. '{print $1}'`"
 ocaml_minor="`echo $OCAMLVERSION | $AWK -F. '{print $2}' | sed 's/^0//'`"
