diff --git i/tests/fprintd.py w/tests/fprintd.py
index 64ea42e..8902955 100644
--- i/tests/fprintd.py
+++ w/tests/fprintd.py
@@ -227,7 +227,7 @@ class FPrintdTest(dbusmock.DBusTestCase):
     def daemon_start(self, driver='Virtual image device for debugging'):
         timeout = get_timeout('daemon_start')  # seconds
         env = os.environ.copy()
-        env['G_DEBUG'] = 'fatal-criticals'
+        env['G_DEBUG'] = ''
         env['STATE_DIRECTORY'] = (self.state_dir + ':' + '/hopefully/a/state_dir_path/that/shouldnt/be/writable')
         env['RUNTIME_DIRECTORY'] = self.run_dir
         # The tests parses the debug output for suspend inhibitor debugging
