Index: webkitgtk-2.38.6/Source/WebCore/platform/graphics/SourceBrush.cpp
===================================================================
--- webkitgtk-2.38.6.orig/Source/WebCore/platform/graphics/SourceBrush.cpp
+++ webkitgtk-2.38.6/Source/WebCore/platform/graphics/SourceBrush.cpp
@@ -65,12 +65,12 @@ Pattern* SourceBrush::pattern() const
 
 void SourceBrush::setGradient(Ref<Gradient>&& gradient, const AffineTransform& spaceTransform)
 {
-    m_brush = { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
+    m_brush = Brush { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
 }
 
 void SourceBrush::setPattern(Ref<Pattern>&& pattern)
 {
-    m_brush = { WTFMove(pattern) };
+    m_brush = Brush { WTFMove(pattern) };
 }
 
 WTF::TextStream& operator<<(TextStream& ts, const SourceBrush& brush)
