Index: libxfce4ui-4.16.1/xfce4-about/system-info.c
===================================================================
--- libxfce4ui-4.16.1.orig/xfce4-about/system-info.c
+++ libxfce4ui-4.16.1/xfce4-about/system-info.c
@@ -69,6 +69,7 @@ static const gchar *const arch_64[] = {
   "aarch64",
   "amd64",
   "ia64",
+  "loongarch64",
   "ppc64",
   "sparc64",
   "x86_64",
@@ -201,7 +202,7 @@ get_cpu_info (const glibtop_sysinfo *inf
   /* count duplicates */
   for (i = 0; i != info->ncpu; ++i)
     {
-      const char * const keys[] = { "model name", "cpu", "Processor" };
+      const char * const keys[] = { "model name", "cpu", "Processor", "Model Name" };
       char *model;
       int  *count;
 
