diff --git i/CMakeLists.txt w/CMakeLists.txt
index d6c16c8..3fa0bf8 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -54,7 +54,7 @@ message( STATUS "Is Synth: ${STOCHAS_IS_SYNTH}, Is Midi Effect: ${STOCHAS_IS_MID
 # I'm not sure at this point whether this has any effect. I know etting it to "Instrument" causes it to show as
 # VSTi in reaper whereas setting it to "Effect" causes it to show as VST
 set(STOCHAS_CATEGORY "Effect") # original was "Instrument"
-add_subdirectory(lib/JUCE)
+find_package(JUCE COMPONENTS REQUIRED)
 
 add_custom_target( git-info BYPRODUCTS ${CMAKE_BINARY_DIR}/geninclude/version.cpp
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
