diff --git a/src/n2n_port_mapping.c b/src/n2n_port_mapping.c
index a5a057f..88b1c34 100644
--- a/src/n2n_port_mapping.c
+++ b/src/n2n_port_mapping.c
@@ -110,7 +110,7 @@ static int n2n_UPNP_GetValidIGD (struct UPNPUrls *urls, struct IGDdatas *data, c
         traceEvent(TRACE_INFO, "    usn: %s", device->usn);
     }
 
-    ret = UPNP_GetValidIGD(devlist, urls, data, lanaddr, N2N_NETMASK_STR_SIZE);
+    ret = UPNP_GetValidIGD(devlist, urls, data, lanaddr, N2N_NETMASK_STR_SIZE, NULL, 0);
     if(ret == 0) {
         traceEvent(TRACE_WARNING, "UPnP get valid IGD failed, code %d (%s)", ret, strupnperror(ret));
         freeUPNPDevlist(devlist);
