diff --git a/src/Views/FirmwareReleaseView.vala b/src/Views/FirmwareReleaseView.vala
index 8e03207c..74411188 100644
--- a/src/Views/FirmwareReleaseView.vala
+++ b/src/Views/FirmwareReleaseView.vala
@@ -214,7 +214,7 @@ public class About.FirmwareReleaseView : Gtk.Grid {
 
         summary_label.label = release.get_summary ();
         try {
-            description_label.label = AppStream.markup_convert_simple (release.get_description ());
+            description_label.label = AppStream.markup_convert (release.get_description (), TEXT);
         } catch (Error e) {
             description_label.label = "";
             warning ("Could not convert markup of release: %s", e.message);
