diff -aur py-stringmatching-0.4.5/py_stringmatching/tests/test_simfunctions.py py-stringmatching-0.4.5.old/py_stringmatching/tests/test_simfunctions.py
--- py-stringmatching-0.4.5/py_stringmatching/tests/test_simfunctions.py	2024-02-01 22:00:33.000000000 +0100
+++ py-stringmatching-0.4.5.old/py_stringmatching/tests/test_simfunctions.py	2024-05-16 09:53:47.808210704 +0200
@@ -3,7 +3,7 @@
 import math
 import unittest
 
-from .utils import raises
+from utils import raises
 
 # sequence based similarity measures
 from py_stringmatching.similarity_measure.affine import Affine
diff -aur py-stringmatching-0.4.5/py_stringmatching/tests/test_sim_Soundex.py py-stringmatching-0.4.5.old/py_stringmatching/tests/test_sim_Soundex.py
--- py-stringmatching-0.4.5/py_stringmatching/tests/test_sim_Soundex.py	2024-02-01 22:00:33.000000000 +0100
+++ py-stringmatching-0.4.5.old/py_stringmatching/tests/test_sim_Soundex.py	2024-05-16 09:54:00.731910749 +0200
@@ -4,7 +4,7 @@
 
 from py_stringmatching.similarity_measure.soundex import Soundex
 
-from .utils import raises
+from utils import raises
 
 
 class SoundexTestCases(unittest.TestCase):
diff -aur py-stringmatching-0.4.5/py_stringmatching/tests/test_tokenizers.py py-stringmatching-0.4.5.old/py_stringmatching/tests/test_tokenizers.py
--- py-stringmatching-0.4.5/py_stringmatching/tests/test_tokenizers.py	2024-02-01 22:00:33.000000000 +0100
+++ py-stringmatching-0.4.5.old/py_stringmatching/tests/test_tokenizers.py	2024-05-16 09:54:02.418625476 +0200
@@ -6,7 +6,7 @@
 from py_stringmatching.tokenizer.qgram_tokenizer import QgramTokenizer
 from py_stringmatching.tokenizer.whitespace_tokenizer import WhitespaceTokenizer
 
-from .utils import raises
+from utils import raises
 
 
 class QgramTokenizerTestCases(unittest.TestCase):
