--- qttools-everywhere-src-6.4.2/src/linguist/lupdate/lupdatepreprocessoraction.h	2023-03-06 23:30:16.999102388 +0800
+++ qttools-everywhere-src-6.4.2/src/linguist/lupdate/lupdatepreprocessoraction.h	2023-03-06 23:32:56.223626722 +0800
@@ -59,7 +59,7 @@
 #endif
                             clang::StringRef /*searchPath*/, clang::StringRef /*relativePath*/,
                             const clang::Module */*imported*/,
-                            clang::SrcMgr::CharacteristicKind /*fileType*/) override;
+                            clang::SrcMgr::CharacteristicKind /*fileType*/);
 
     std::string m_inputFile;
     clang::Preprocessor &m_preprocessor;
