diff --git a/userspace/libsinsp/CMakeLists.txt b/userspace/libsinsp/CMakeLists.txt
index e75ceb3..4bcbd5a 100644
--- a/userspace/libsinsp/CMakeLists.txt
+++ b/userspace/libsinsp/CMakeLists.txt
@@ -38,6 +38,7 @@ include(ExternalProject)
 
 if(NOT WIN32 AND NOT APPLE)
 	include(b64)
+	find_library(ZSTD_LIB NAMES zstd)
 endif()
 
 include(jsoncpp)
@@ -292,7 +293,8 @@ if(NOT WIN32)
 						"${GPR_LIB}"
 						"${PROTOBUF_LIB}"
 						"${CARES_LIB}"
-						"${JQ_LIB}")
+						"${JQ_LIB}"
+						"${ZSTD_LIB}")
 
 			if(NOT MUSL_OPTIMIZED_BUILD)
 				target_link_libraries(sinsp INTERFACE rt anl)
