diff --git a/gb.qt4/src/canimation.h b/gb.qt4/src/canimation.h
index 681b92938..ed2a11213 100644
--- a/gb.qt4/src/canimation.h
+++ b/gb.qt4/src/canimation.h
@@ -24,11 +24,11 @@
 #ifndef __CANIMATION_H
 #define __CANIMATION_H
 
-#include "gambas.h"
-
 #include <QMovie>
 #include <QBuffer>
 
+#include "gambas.h"
+
 #ifndef __CANIMATION_CPP
 extern GB_DESC AnimationDesc[];
 #else
diff --git a/gb.qt5/src/webview/c_websettings.h b/gb.qt5/src/webview/c_websettings.h
index bca29f369..d65b3ea84 100644
--- a/gb.qt5/src/webview/c_websettings.h
+++ b/gb.qt5/src/webview/c_websettings.h
@@ -24,12 +24,12 @@
 #ifndef __C_WEBSETTINGS_H
 #define __C_WEBSETTINGS_H
 
-#include "main.h"
-
 #include <QUrl>
 #include <QWebEngineView>
 #include <QWebEngineSettings>
 
+#include "main.h"
+
 typedef
   struct {
     GB_BASE ob;
