Index: rustc-1.63.0-src/src/bootstrap/bootstrap.py
===================================================================
--- rustc-1.63.0-src.orig/src/bootstrap/bootstrap.py
+++ rustc-1.63.0-src/src/bootstrap/bootstrap.py
@@ -796,8 +796,6 @@ class RustBuild(object):
             args.append("--verbose")
         if self.use_locked_deps:
             args.append("--locked")
-        if self.use_vendored_sources:
-            args.append("--frozen")
         if self.get_toml("metrics", "build"):
             args.append("--features")
             args.append("build-metrics")
Index: rustc-1.63.0-src/src/bootstrap/builder.rs
===================================================================
--- rustc-1.63.0-src.orig/src/bootstrap/builder.rs
+++ rustc-1.63.0-src/src/bootstrap/builder.rs
@@ -2052,9 +2052,6 @@ impl<'a> Builder<'a> {
         if self.config.locked_deps {
             cargo.arg("--locked");
         }
-        if self.config.vendor || self.is_sudo {
-            cargo.arg("--frozen");
-        }
 
         // Try to use a sysroot-relative bindir, in case it was configured absolutely.
         cargo.env("RUSTC_INSTALL_BINDIR", self.config.bindir_relative());
