commit 81880596c333426fc09b8f45160b27d514f1afdd
Author: Xiaotian Wu <wuxiaotian@loongson.cn>
Date:   Fri Sep 23 13:02:41 2022 +0800

    fix features differ

diff --git a/src/bootstrap/tool.rs b/src/bootstrap/tool.rs
index 905fa431d29..2766083651f 100644
--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -194,7 +194,7 @@ fn run(self, builder: &Builder<'_>) -> Option<PathBuf> {
                       that will update the dependency graph to ensure that \
                       these crates all share the same feature set"
             );
-            panic!("tools should not compile multiple copies of the same crate");
+            //panic!("tools should not compile multiple copies of the same crate");
         }
 
         builder.save_toolstate(
