--- a/simple_term_menu.py	2023-07-19 22:41:08.000000000 +0800
+++ b/tmp/simple_term_menu.py	2023-08-15 14:44:22.410664526 +0800
@@ -1270,7 +1270,7 @@
                             BoxDrawingCharacters.upper_left
                             + (2 * BoxDrawingCharacters.horizontal + " " + self._preview_title)[: num_cols - 3]
                             + " "
-                            + (num_cols - len(self._preview_title) - 6) * BoxDrawingCharacters.horizontal
+                            + (num_cols - wcswidth(self._preview_title) - 6) * BoxDrawingCharacters.horizontal
                             + BoxDrawingCharacters.upper_right
                         )[:num_cols]
                         + "\n"
