diff --git a/registeredQDbus/sysdbusregister.cpp b/registeredQDbus/sysdbusregister.cpp
index 4b7786f0..2e69c2ac 100644
--- a/registeredQDbus/sysdbusregister.cpp
+++ b/registeredQDbus/sysdbusregister.cpp
@@ -339,7 +339,7 @@ void SysdbusRegister::_getDisplayInfoThread()
                     DDCA_Display_Ref ddca_dref;
                     display.edidHash = edidHash;
                     ddca_create_edid_display_identifier(dlist_loc->info[i].edid_bytes,&did);
-                    ddca_create_display_ref(did,&ddca_dref);
+                    ddca_get_display_ref(did,&ddca_dref);
                     ddca_open_display2(ddca_dref,false,&display.ddca_dh_loc);
                     displayInfo_V.append(display);
                 }
