diff --git i/libs/aaf/wscript w/libs/aaf/wscript
index 5ca7958a17..a43d94f8e4 100644
--- i/libs/aaf/wscript
+++ w/libs/aaf/wscript
@@ -37,13 +37,9 @@ def options(opt):
     autowaf.set_options(opt)
 
 def configure(conf):
-    if conf.is_defined('USE_EXTERNAL_LIBS'):
-        autowaf.check_pkg(conf, 'aaf', uselib_store='LIBAAF', mandatory=True)
+    return
 
 def build(bld):
-    if bld.is_defined('USE_EXTERNAL_LIBS'):
-        return
-
     if bld.is_defined ('INTERNAL_SHARED_LIBS'):
         obj              = bld.shlib(features = 'c cshlib', source=libaaf_sources)
         obj.defines = [ 'LIBAAF_DLL_EXPORTS=1' ]
diff --git i/session_utils/wscript w/session_utils/wscript
index b7f6ef2b19..d4120a47b0 100644
--- i/session_utils/wscript
+++ w/session_utils/wscript
@@ -45,10 +45,7 @@ def build_ardour_util(bld, util):
     obj.uselib       += ' FFTW3F LO TAGLIB LILV RUBBERBAND AUBIO LRDF ARCHIVE VAMPSDK VAMPHOSTSDK'
 
     if util == "new_aaf_session":
-        if bld.is_defined('USE_EXTERNAL_LIBS'):
-            obj.uselib += ' LIBAAF'
-        else:
-            obj.use.extend (['libaaf'])
+        obj.use.extend (['libaaf'])
 
     if bld.is_defined('HAVE_SUIL'):
         obj.uselib += ' SUIL'
