diff --git a/libs/ardour/audiofile_tagger.cc b/libs/ardour/audiofile_tagger.cc
index 591b8e9238..47b2433936 100644
--- a/libs/ardour/audiofile_tagger.cc
+++ b/libs/ardour/audiofile_tagger.cc
@@ -34,6 +34,8 @@
 #include <taglib/taglib.h>
 #include <taglib/xiphcomment.h>
 
+#include <iostream>
+
 /* Convert string to TagLib::String */
 #define TL_STR(string) TagLib::String ((string).c_str(), TagLib::String::UTF8)
 
