diff -aur rawkit-0.6.0.old/requirements-dev.txt rawkit-0.6.0/requirements-dev.txt
--- rawkit-0.6.0.old/requirements-dev.txt	2024-02-10 14:49:36.465514950 +0100
+++ rawkit-0.6.0/requirements-dev.txt	2024-02-10 14:50:58.400543550 +0100
@@ -2,7 +2,6 @@
 
 coverage
 flake8
-mock
 sphinx
 pytest
 wheel
diff -aur rawkit-0.6.0.old/tests/unit/bindings_test.py rawkit-0.6.0/tests/unit/bindings_test.py
--- rawkit-0.6.0.old/tests/unit/bindings_test.py	2024-02-10 14:49:36.465514950 +0100
+++ rawkit-0.6.0/tests/unit/bindings_test.py	2024-02-10 14:50:51.387065007 +0100
@@ -1,5 +1,5 @@
 import ctypes
-import mock
+from unittest import mock
 import pytest
 
 from libraw.bindings import LibRaw
diff -aur rawkit-0.6.0.old/tests/unit/options_test.py rawkit-0.6.0/tests/unit/options_test.py
--- rawkit-0.6.0.old/tests/unit/options_test.py	2024-02-10 14:49:36.465514950 +0100
+++ rawkit-0.6.0/tests/unit/options_test.py	2024-02-10 14:50:47.093642788 +0100
@@ -1,6 +1,6 @@
 import pytest
 
-from mock import Mock
+from unittest.mock import Mock
 from rawkit.options import option, Options, WhiteBalance
 
 
diff -aur rawkit-0.6.0.old/tests/unit/raw_test.py rawkit-0.6.0/tests/unit/raw_test.py
--- rawkit-0.6.0.old/tests/unit/raw_test.py	2024-02-10 14:49:36.465514950 +0100
+++ rawkit-0.6.0/tests/unit/raw_test.py	2024-02-10 14:50:40.856847013 +0100
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 import os
 import pytest
 import warnings
diff -aur rawkit-0.6.0.old/tests/unit/util_test.py rawkit-0.6.0/tests/unit/util_test.py
--- rawkit-0.6.0.old/tests/unit/util_test.py	2024-02-10 14:49:36.465514950 +0100
+++ rawkit-0.6.0/tests/unit/util_test.py	2024-02-10 14:50:34.410046914 +0100
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 import pytest
 
 from rawkit import util
