diff --git a/tests/conftest.py b/tests/conftest.py
index 770bd9c..c9dd043 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -167,27 +167,6 @@ def skip_by_platform(request: pytest.FixtureRequest) -> None:
         pytest.skip(f"skipped on this platform: {sys.platform}")
 
 
-def pytest_addoption(parser: pytest.Parser) -> None:
-    group = parser.getgroup("playwright", "Playwright")
-    group.addoption(
-        "--browser",
-        action="append",
-        default=[],
-        help="Browsers which should be used. By default on all the browsers.",
-    )
-    group.addoption(
-        "--browser-channel",
-        action="store",
-        default=None,
-        help="Browser channel to be used.",
-    )
-    parser.addoption(
-        "--headed",
-        action="store_true",
-        default=False,
-        help="Run tests in headed mode.",
-    )
-
 
 @pytest.fixture(scope="session")
 def assert_to_be_golden(browser_name: str) -> Callable[[bytes, str], None]:
