diff -ur signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/main.cpp signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/main.cpp
--- signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/main.cpp	2023-04-05 13:01:45.048480875 +0800
+++ signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/main.cpp	2017-10-23 23:39:24.000000000 +0800
@@ -29,7 +29,6 @@
 #include <QDBusConnection>
 #include <QProcessEnvironment>
 #include <QSettings>
-#include <QtWebEngine>
 
 using namespace SignOnUi;
 
@@ -39,7 +40,6 @@
 int main(int argc, char **argv)
 {
     QApplication app(argc, argv);
-    QtWebEngine::initialize();
 
     app.setApplicationName("signon-ui");
     app.setQuitOnLastWindowClosed(false);
diff -ur signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/signon-ui.pro signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/signon-ui.pro
--- signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/signon-ui.pro	2023-04-05 13:00:36.856326177 +0800
+++ signon-ui-4368bb77d9d1abc2978af514225ba4a42c29a646/src/signon-ui.pro	2017-10-23 23:39:24.000000000 +0800
@@ -17,7 +17,6 @@
     gui \
     network \
-    quick \
-    webengine
+    quick
 
 PKGCONFIG += \
     signon-plugins-common \
