--- a/contrib/soju.service
+++ b/contrib/soju.service
@@ -9,12 +9,9 @@ After=network-online.target
 Type=simple
 User=soju
 Group=soju
-DynamicUser=yes
-StateDirectory=soju
-ConfigurationDirectory=soju
 RuntimeDirectory=soju
 AmbientCapabilities=CAP_NET_BIND_SERVICE
-ExecStart=/usr/bin/soju
+ExecStart=/usr/bin/soju --config /etc/soju/config
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
 
