diff -ru dde-file-manager-6.0.37.orig/src/plugins/filemanager/core/dfmplugin-workspace/views/private/iconitemeditor_p.h dde-file-manager-6.0.37/src/plugins/filemanager/core/dfmplugin-workspace/views/private/iconitemeditor_p.h
--- dde-file-manager-6.0.37.orig/src/plugins/filemanager/core/dfmplugin-workspace/views/private/iconitemeditor_p.h	2024-01-24 19:43:00.299274817 +0100
+++ dde-file-manager-6.0.37/src/plugins/filemanager/core/dfmplugin-workspace/views/private/iconitemeditor_p.h	2024-01-24 19:55:18.796878388 +0100
@@ -7,6 +7,7 @@
 #include <DArrowRectangle>
 
 #include <QStack>
+#include <QGraphicsEffect>
 
 class QTextEdit;
 
