diff -ru dde-file-manager-6.0.37.orig/src/plugins/common/dfmplugin-preview/pluginpreviews/music-preview/musicmessageview.cpp dde-file-manager-6.0.37/src/plugins/common/dfmplugin-preview/pluginpreviews/music-preview/musicmessageview.cpp
--- dde-file-manager-6.0.37.orig/src/plugins/common/dfmplugin-preview/pluginpreviews/music-preview/musicmessageview.cpp	2024-01-24 19:43:00.259252816 +0100
+++ dde-file-manager-6.0.37/src/plugins/common/dfmplugin-preview/pluginpreviews/music-preview/musicmessageview.cpp	2024-01-24 19:54:27.853252696 +0100
@@ -239,9 +239,9 @@
 {
     TagLib::Tag *tag = static_cast<TagLib::Tag *>(obj);
     bool encode = true;
-    encode &= tag->title().isNull() ? true : tag->title().isLatin1();
-    encode &= tag->artist().isNull() ? true : tag->artist().isLatin1();
-    encode &= tag->album().isNull() ? true : tag->album().isLatin1();
+    encode &= tag->title().isEmpty() ? true : tag->title().isLatin1();
+    encode &= tag->artist().isEmpty() ? true : tag->artist().isLatin1();
+    encode &= tag->album().isEmpty() ? true : tag->album().isLatin1();
 
     QByteArray detectByte;
     QByteArray detectCodec;
