--- a/build/src/github.com/bugsnag/panicwrap/dup2.go	2019-09-02 19:53:51.000000000 +0800
+++ b/build/src/github.com/bugsnag/panicwrap/dup2.go	2022-11-27 21:10:31.367788962 +0800
@@ -1,4 +1,4 @@
-// +build darwin dragonfly freebsd linux,!arm64 netbsd openbsd
+// +build darwin dragonfly freebsd linux,!arm64 linux,!loong64 netbsd openbsd
 
 package panicwrap
 
--- a/build/src/github.com/bugsnag/panicwrap/dup3.go	2019-09-02 19:53:51.000000000 +0800
+++ b/build/src/github.com/bugsnag/panicwrap/dup3.go	2022-11-27 21:04:11.457811920 +0800
@@ -1,4 +1,4 @@
-// +build linux,arm64
+// +build linux,arm64 linux,loong64
 
 package panicwrap
 
