diff -ruN a/sigal/image.py b/sigal/image.py
--- a/sigal/image.py	2022-04-08 20:19:56.000000000 +0200
+++ b/sigal/image.py	2023-07-06 12:23:59.802110427 +0200
@@ -162,9 +162,9 @@
     original_format = img.format
 
     if fit:
-        img = ImageOps.fit(img, box, PILImage.ANTIALIAS, centering=thumb_fit_centering)
+        img = ImageOps.fit(img, box, PILImage.LANCZOS, centering=thumb_fit_centering)
     else:
-        img.thumbnail(box, PILImage.ANTIALIAS)
+        img.thumbnail(box, PILImage.LANCZOS)
 
     outformat = img.format or original_format or 'JPEG'
     logger.debug('Save thumnail image: %s (%s)', outname, outformat)
