diff -aur pytest_container-0.2.0/tests/conftest.py pytest_container-0.2.0.new/tests/conftest.py
--- pytest_container-0.2.0/tests/conftest.py	2023-06-18 15:04:05.000000000 +0200
+++ pytest_container-0.2.0.new/tests/conftest.py	2023-06-30 17:41:51.115059224 +0200
@@ -1,6 +1,9 @@
 # pylint: disable=missing-function-docstring,missing-module-docstring
 from typeguard import typechecked
-from typeguard.importhook import install_import_hook
+try:
+    from typeguard.importhook import install_import_hook
+except ImportError:
+    from typeguard import install_import_hook
 
 from pytest_container import add_extra_run_and_build_args_options
 from pytest_container import add_logging_level_options
