--- HIPIFY-rocm-5.6.0/CMakeLists.txt.bak	2023-07-07 18:01:35.058007323 +0200
+++ HIPIFY-rocm-5.6.0/CMakeLists.txt	2023-07-07 18:01:48.627981805 +0200
@@ -134,19 +134,6 @@
     USE_SOURCE_PERMISSIONS
     PATTERN "hipify-perl"
     PATTERN "*.sh")
-# install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path
-install(
-    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
-    DESTINATION .
-    COMPONENT clang-resource-headers
-    FILES_MATCHING
-    PATTERN "*.h"
-    PATTERN "*.modulemap"
-    PATTERN "algorithm"
-    PATTERN "complex"
-    PATTERN "new"
-    PATTERN "ppc_wrappers" EXCLUDE
-    PATTERN "openmp_wrappers" EXCLUDE)
 
 option(FILE_REORG_BACKWARD_COMPATIBILITY "Enable File Reorg with backward compatibility" ON)
 
