--- HIPIFY-rocm-5.7.0/CMakeLists.txt.bak	2023-10-08 14:40:20.247520858 +0200
+++ HIPIFY-rocm-5.7.0/CMakeLists.txt	2023-10-08 14:41:13.315283175 +0200
@@ -162,37 +162,6 @@
     PATTERN "hipify-perl"
     PATTERN "*.sh")
 
-  # Headers are already included in HIP SDK, so skip those if including HIPIFY in HIP SDK.
-  if(NOT HIPIFY_INCLUDE_IN_HIP_SDK)
-    # Install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path.
-    install(
-      DIRECTORY ${LLVM_DIR}/../../clang/${LIB_CLANG_RES}/
-      DESTINATION .
-      COMPONENT clang-resource-headers
-      FILES_MATCHING
-      PATTERN "*.h"
-      PATTERN "*.modulemap"
-      PATTERN "algorithm"
-      PATTERN "complex"
-      PATTERN "new"
-      PATTERN "ppc_wrappers" EXCLUDE
-      PATTERN "openmp_wrappers" EXCLUDE)
-  endif()
-
-# install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path
-install(
-    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
-    DESTINATION .
-    COMPONENT clang-resource-headers
-    FILES_MATCHING
-    PATTERN "*.h"
-    PATTERN "*.modulemap"
-    PATTERN "algorithm"
-    PATTERN "complex"
-    PATTERN "new"
-    PATTERN "ppc_wrappers" EXCLUDE
-    PATTERN "openmp_wrappers" EXCLUDE)
-
   if(UNIX)
     # Get rid of any RPATH definations already.
     set_target_properties(hipify-clang PROPERTIES INSTALL_RPATH "")
