--- rocprofiler-rocm-5.7.1/CMakeLists.txt.bak	2023-11-10 15:14:09.927643993 +0100
+++ rocprofiler-rocm-5.7.1/CMakeLists.txt	2023-11-10 15:14:14.575626266 +0100
@@ -160,14 +160,6 @@
 # Build libraries
 add_subdirectory(src)
 
-if(${LIBRARY_TYPE} STREQUAL SHARED)
-    # Build samples
-    add_subdirectory(samples)
-
-    # Build tests
-    add_subdirectory(tests-v2)
-endif()
-
 # Build Plugins
 add_subdirectory(plugin)
 
--- rocprofiler-rocm-5.7.1/test/CMakeLists.txt.bak	2023-11-10 15:17:59.908912199 +0100
+++ rocprofiler-rocm-5.7.1/test/CMakeLists.txt	2023-11-10 15:18:13.978874501 +0100
@@ -112,10 +112,6 @@
         DEPENDS ${INPUT_FILE} clang
         COMMENT "Building ${OUTPUT_FILE}..."
         VERBATIM)
-    install(
-        FILES ${PROJECT_BINARY_DIR}/${OUTPUT_FILE}
-        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tests-v1
-        COMPONENT tests)
     set(HSACO_TARGET_LIST
         ${HSACO_TARGET_LIST} ${PROJECT_BINARY_DIR}/${OUTPUT_FILE}
         PARENT_SCOPE)
