diff --git a/meson.build b/meson.build
index eb3a2b8..e8b824d 100644
--- a/meson.build
+++ b/meson.build
@@ -13,6 +13,7 @@ project(
 
 add_project_arguments('-Wno-unused-parameter', language: 'c')
 add_project_arguments('-Wno-missing-braces', language: 'c')
+add_project_arguments('-Wno-deprecated-declarations', language: 'c')
 
 subdir('gtkgreet')
 
@@ -41,4 +42,4 @@ if scdoc.found()
 			install_dir: '@0@/man@1@'.format(mandir, section)
 		)
 	endforeach
-endif
\ No newline at end of file
+endif
