diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0cc425a..0bda1201 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -556,7 +556,7 @@ if(NCNN_VULKAN)
             message(FATAL_ERROR "The submodules were not downloaded! Please update submodules with \"git submodule update --init\" and try again.")
         else()
             # glslang requires c++11
-            set(CMAKE_CXX_STANDARD 11)
+            set(CMAKE_CXX_STANDARD 17)
 
             option(BUILD_EXTERNAL "" OFF)
             option(ENABLE_SPVREMAPPER "" OFF)
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index a7739be2..02a54afe 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -35,7 +35,7 @@ if(NCNN_PIXEL)
             message(STATUS "    include path: ${OpenCV_INCLUDE_DIRS}")
 
             if(${OpenCV_VERSION_MAJOR} GREATER 3)
-                set(CMAKE_CXX_STANDARD 11)
+                set(CMAKE_CXX_STANDARD 17)
             endif()
         endif()
 
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 999efa1d..87236c15 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -5,7 +5,7 @@ project(pyncnn)
 set(PACKAGE_VERSION ${NCNN_VERSION_STRING})
 add_definitions(-DVERSION_INFO="${PACKAGE_VERSION}")
 
-set( CMAKE_CXX_STANDARD 11 )
+set( CMAKE_CXX_STANDARD 17 )
 set( CMAKE_CXX_STANDARD_REQUIRED ON )
 
 if(CMAKE_CXX_COMPILER_ARCHITECTURE_ID MATCHES "ARM64")
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 965046be..5aef5a4a 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.1) # for CMAKE_CXX_STANDARD
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
 
 include(GNUInstallDirs)
 macro(ncnn_install_tool toolname)
diff --git a/tools/caffe/CMakeLists.txt b/tools/caffe/CMakeLists.txt
index 733ec1a6..e9142ca4 100644
--- a/tools/caffe/CMakeLists.txt
+++ b/tools/caffe/CMakeLists.txt
@@ -1,7 +1,9 @@
 
-find_package(Protobuf)
+find_package(protobuf CONFIG)
+find_package(Protobuf) # for protobuf_generate_cpp
+set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
 
-if(PROTOBUF_FOUND)
+if(protobuf_FOUND)
     protobuf_generate_cpp(CAFFE_PROTO_SRCS CAFFE_PROTO_HDRS caffe.proto)
     add_executable(caffe2ncnn caffe2ncnn.cpp ${CAFFE_PROTO_SRCS} ${CAFFE_PROTO_HDRS})
     target_include_directories(caffe2ncnn
diff --git a/tools/onnx/CMakeLists.txt b/tools/onnx/CMakeLists.txt
index f78c3f8e..bf14d756 100644
--- a/tools/onnx/CMakeLists.txt
+++ b/tools/onnx/CMakeLists.txt
@@ -1,7 +1,8 @@
 
-find_package(Protobuf)
+find_package(protobuf CONFIG)
+set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
 
-if(PROTOBUF_FOUND)
+if(protobuf_FOUND)
     protobuf_generate_cpp(ONNX_PROTO_SRCS ONNX_PROTO_HDRS onnx.proto)
     add_executable(onnx2ncnn onnx2ncnn.cpp ${ONNX_PROTO_SRCS} ${ONNX_PROTO_HDRS})
     target_include_directories(onnx2ncnn
diff --git a/tools/pnnx/src/CMakeLists.txt b/tools/pnnx/src/CMakeLists.txt
index 7ec9fc08..b9bcf676 100644
--- a/tools/pnnx/src/CMakeLists.txt
+++ b/tools/pnnx/src/CMakeLists.txt
@@ -532,8 +532,9 @@ set(pnnx_pass_ncnn_SRCS
     pass_ncnn/torchvision_DeformConv2d.cpp
 )
 
-find_package(Protobuf)
-if(PROTOBUF_FOUND)
+find_package(protobuf CONFIG)
+set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
+if(protobuf_FOUND)
     protobuf_generate_cpp(ONNX_PROTO_SRCS ONNX_PROTO_HDRS onnx.proto)
 
     add_library(pnnx2onnx STATIC
