--- mariadb-10.11.2/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h	2023-03-06 22:26:59.796119232 +0800
+++ mariadb-10.11.2/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h	2023-03-06 22:27:26.922551733 +0800
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include "rocksdb/rocksdb_namespace.h"
 
 struct CompactionIterationStats {
--- mariadb-10.11.2/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.cc	2023-03-06 22:58:47.099159445 +0800
+++ mariadb-10.11.2/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.cc	2023-03-06 22:58:57.630503658 +0800
@@ -4,6 +4,7 @@
 //  (found in the LICENSE.Apache file in the root directory).
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "rocksdb/slice.h"
 #include "table/block_based/data_block_hash_index.h"
--- mariadb-10.11.2/storage/rocksdb/rocksdb/util/string_util.h	2023-03-06 23:16:58.158923011 +0800
+++ mariadb-10.11.2/storage/rocksdb/rocksdb/util/string_util.h	2023-03-06 23:17:07.982758034 +0800
@@ -10,6 +10,7 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <cstdint>
 
 #include "rocksdb/rocksdb_namespace.h"
 
--- mariadb-10.11.2/storage/rocksdb/rocksdb/utilities/checkpoint/checkpoint_impl.h	2023-03-06 23:31:25.919907545 +0800
+++ mariadb-10.11.2/storage/rocksdb/rocksdb/utilities/checkpoint/checkpoint_impl.h	2023-03-06 23:31:39.587266180 +0800
@@ -9,6 +9,7 @@
 #include "rocksdb/utilities/checkpoint.h"
 
 #include <string>
+#include <cstdint>
 #include "file/filename.h"
 #include "rocksdb/db.h"
 
--- mariadb-10.11.2/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h	2023-03-07 15:08:08.867287813 +0800
+++ mariadb-10.11.2/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h	2023-03-07 15:08:17.830792027 +0800
@@ -10,6 +10,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 #include "rocksdb/status.h"
 
 namespace ROCKSDB_NAMESPACE {
