--- a/adriconf/Utils/PCIDatabaseQueryInterface.h
+++ b/adriconf/Utils/PCIDatabaseQueryInterface.h
@@ -1,6 +1,7 @@
 #ifndef ADRICONF_PCIDATABASEQUERYINTERFACE_H
 #define ADRICONF_PCIDATABASEQUERYINTERFACE_H
 
+#include <cstdint>
 #include <glibmm/ustring.h>
 
 class PCIDatabaseQueryInterface {
@@ -11,4 +12,4 @@ public:
     virtual Glib::ustring queryDeviceName(uint16_t vendorId, uint16_t deviceId) = 0;
 };
 
-#endif //ADRICONF_PCIDATABASEQUERYINTERFACE_H
\ No newline at end of file
+#endif //ADRICONF_PCIDATABASEQUERYINTERFACE_H
