diff --git a/test/unit/module/maintenance/test_update_documentation.py b/test/unit/module/maintenance/test_update_documentation.py
index 2feca2e3..5bc1d10c 100644
--- a/test/unit/module/maintenance/test_update_documentation.py
+++ b/test/unit/module/maintenance/test_update_documentation.py
@@ -3,6 +3,7 @@ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 SPDX-License-Identifier: MIT-0
 """
 import logging
+import os.path
 import sys
 from test.testlib.testcase import BaseTestCase
 from unittest.mock import call, mock_open, patch
@@ -23,6 +24,9 @@ Regular Text
 """
 
     def test_update_docs(self):
+        if not os.path.exists('.git'):
+            self.skipTest('update_documentation() function needs a git repo')
+
         collection = RulesCollection(include_rules=["I"], include_experimental=True)
 
         builtin_module_name = "builtins"
