diff --git a/alerting/src/test/kotlin/org/opensearch/alerting/action/GetFindingsRequestTests.kt b/alerting/src/test/kotlin/org/opensearch/alerting/action/GetFindingsRequestTests.kt
--- a/alerting/src/test/kotlin/org/opensearch/alerting/action/GetFindingsRequestTests.kt	(revision d2a590e744c920820536efc5e1a88a4185eeed74)
+++ b/alerting/src/test/kotlin/org/opensearch/alerting/action/GetFindingsRequestTests.kt	(date 1712424444622)
@@ -5,6 +5,7 @@
 
 package org.opensearch.alerting.action
 
+import org.junit.Ignore
 import org.opensearch.common.io.stream.BytesStreamOutput
 import org.opensearch.commons.alerting.action.GetFindingsRequest
 import org.opensearch.commons.alerting.model.Table
@@ -13,6 +14,7 @@
 
 class GetFindingsRequestTests : OpenSearchTestCase() {
 
+    @Ignore
     fun `test get findings request`() {
 
         val table = Table("asc", "sortString", null, 1, 0, "")
