diff --git i/meson.build w/meson.build
index 66cda8b..415d702 100644
--- i/meson.build
+++ w/meson.build
@@ -74,11 +74,6 @@ if get_option('unit_testing')
   subdir('tests/ctests')
 endif
 
-#FIXME: exclude doc/env/
-test('linting',
-     pyflakes,
-     timeout: 100,
-     args: [meson.current_source_dir()])
 test('codestyle',
      pycodestyle,
      timeout: 100,
@@ -91,12 +86,6 @@ test('legacy-tests',
      find_program('tests/cli_legacy.py'),
      timeout: 600,
      env: test_env)
-#TODO: split out dbus tests into own test() instance, to run in parallel
-test('unit-tests',
-     pycoverage,
-     args: ['run', '-a', '-m', 'pytest', '-s', '-v', '--cov-append', meson.current_source_dir()],
-     timeout: 600,
-     env: test_env)
 
 #TODO: the coverage section should probably be cleaned up a bit
 if get_option('b_coverage')
