Index: netplan-0.106/Makefile
===================================================================
--- netplan-0.106.orig/Makefile
+++ netplan-0.106/Makefile
@@ -67,7 +67,7 @@ PYCODESTYLE3 ?= $(shell command -v pycod
 PYTEST3 ?= $(shell command -v pytest-3 || command -v pytest3 || echo true)
 PYCOVERAGE ?= $(shell command -v python3-coverage || echo true)
 
-default: netplan/_features.py generate netplan-dbus dbus/io.netplan.Netplan.service doc/netplan.html doc/netplan.5 doc/netplan-generate.8 doc/netplan-apply.8 doc/netplan-try.8 doc/netplan-dbus.8 doc/netplan-get.8 doc/netplan-set.8
+default: netplan/_features.py generate netplan-dbus dbus/io.netplan.Netplan.service #doc/netplan.html doc/netplan.5 doc/netplan-generate.8 doc/netplan-apply.8 doc/netplan-try.8 doc/netplan-dbus.8 doc/netplan-get.8 doc/netplan-set.8
 
 %.o: src/%.c src/_features.h
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -c $^ `pkg-config --cflags --libs glib-2.0 gio-2.0 yaml-0.1 uuid`
@@ -160,10 +160,10 @@ install: default
 	install -m 644 include/*.h $(DESTDIR)/$(INCLUDEDIR)/netplan/
 	# TODO: install pkg-config once available
 	# docs, data
-	install -m 644 doc/*.html $(DESTDIR)/$(DOCDIR)/netplan/
+	#install -m 644 doc/*.html $(DESTDIR)/$(DOCDIR)/netplan/
 	install -m 644 examples/*.yaml $(DESTDIR)/$(DOCDIR)/netplan/examples/
-	install -m 644 doc/*.5 $(DESTDIR)/$(MANDIR)/man5/
-	install -m 644 doc/*.8 $(DESTDIR)/$(MANDIR)/man8/
+	#install -m 644 doc/*.5 $(DESTDIR)/$(MANDIR)/man5/
+	#install -m 644 doc/*.8 $(DESTDIR)/$(MANDIR)/man8/
 	install -T -D -m 644 netplan.completions $(DESTDIR)/$(BASH_COMPLETIONS_DIR)/netplan
 	# dbus
 	mkdir -p $(DESTDIR)/$(DATADIR)/dbus-1/system.d $(DESTDIR)/$(DATADIR)/dbus-1/system-services
