diff --git a/src/update.c b/src/update.c
index 7a225b5..b2d84b7 100644
--- a/src/update.c
+++ b/src/update.c
@@ -628,7 +628,11 @@ int pkgfile_update(struct repovec_t *repos, struct config_t *config) {
   if (repos->architecture == NULL) {
     struct utsname un;
     uname(&un);
-    repos->architecture = strdup(un.machine);
+    if (strcmp(un.machine, "loongarch64") == 0) {
+      repos->architecture = strdup("loong64");
+    } else {
+      repos->architecture = strdup(un.machine);
+    }
   }
 
   /* ensure all our DBs are 0644 */
