Index: psiconv-0.9.9/program/psiconv/gen_html4.c
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/gen_html4.c
+++ psiconv-0.9.9/program/psiconv/gen_html4.c
@@ -32,6 +32,8 @@
 
 #define TEMPSTR_LEN 100
 
+static psiconv_list fileformat_list; /* of struct psiconv_fileformat */
+
 static void text(const psiconv_config config,psiconv_list list,
                  psiconv_string_t data,const encoding enc);
 static void header(const psiconv_config config, psiconv_list list, 
Index: psiconv-0.9.9/program/psiconv/gen_image.c
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/gen_image.c
+++ psiconv-0.9.9/program/psiconv/gen_image.c
@@ -33,6 +33,7 @@
 #endif
 
 #ifdef IMAGEMAGICK
+static psiconv_list fileformat_list; /* of struct psiconv_fileformat */
 static Image *get_paint_data_section(psiconv_paint_data_section sec);
 static void image_to_list(psiconv_list list,Image *image,const char *dest);
 static void gen_image_list(const psiconv_config config,psiconv_list list,
Index: psiconv-0.9.9/program/psiconv/gen_txt.c
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/gen_txt.c
+++ psiconv-0.9.9/program/psiconv/gen_txt.c
@@ -31,6 +31,7 @@
 #ifdef DMALLOC
 #include "dmalloc.h"
 #endif
+static psiconv_list fileformat_list; /* of struct psiconv_fileformat */
 
 static void output_para(const psiconv_config config,psiconv_list list,
                         const psiconv_paragraph para,encoding encoding_type);
Index: psiconv-0.9.9/program/psiconv/gen_xhtml.c
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/gen_xhtml.c
+++ psiconv-0.9.9/program/psiconv/gen_xhtml.c
@@ -33,6 +33,8 @@
 
 #define TEMPSTR_LEN 100
 
+static psiconv_list fileformat_list; /* of struct psiconv_fileformat */
+
 static void text(const psiconv_config config,psiconv_list list,
           psiconv_string_t data,const encoding enc);
 static void color(const psiconv_config config, psiconv_list list,
Index: psiconv-0.9.9/program/psiconv/psiconv.c
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/psiconv.c
+++ psiconv-0.9.9/program/psiconv/psiconv.c
@@ -47,6 +47,7 @@
 static void print_help(void);
 static void print_version(void);
 static void strtoupper(char *str);
+static psiconv_list fileformat_list; /* of struct psiconv_fileformat */
 
 void print_help(void)
 {
Index: psiconv-0.9.9/program/psiconv/psiconv.h
===================================================================
--- psiconv-0.9.9.orig/program/psiconv/psiconv.h
+++ psiconv-0.9.9/program/psiconv/psiconv.h
@@ -52,7 +52,5 @@ typedef struct fileformat_s {
   output_function *output;
 } *fileformat;
 
-psiconv_list fileformat_list; /* of struct psiconv_fileformat */
-
 
 #endif /* PSICONV_H */
