From 13c80e8ca5ad9f46659c1c29908c9ffac77fb0de Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@v-signal.xyz>
Date: Sun, 3 Jul 2022 10:50:02 +0800
Subject: [PATCH 1/2] add loongarch support

---
 src/tbb/tools_api/ittnotify_config.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/tbb/tools_api/ittnotify_config.h b/src/tbb/tools_api/ittnotify_config.h
index f904a8e9d..fe2a349f8 100644
--- a/src/tbb/tools_api/ittnotify_config.h
+++ b/src/tbb/tools_api/ittnotify_config.h
@@ -163,6 +163,10 @@
 #  define ITT_ARCH_ARM64  6
 #endif /* ITT_ARCH_ARM64 */
 
+#ifndef ITT_ARCH_LOONGARCH64
+#  define ITT_ARCH_LOONGARCH64  7
+#endif /* ITT_ARCH_LOONGARCH64 */
+
 #ifndef ITT_ARCH
 #  if defined _M_IX86 || defined __i386__
 #    define ITT_ARCH ITT_ARCH_IA32
@@ -176,6 +180,8 @@
 #    define ITT_ARCH ITT_ARCH_ARM64
 #  elif defined __powerpc64__
 #    define ITT_ARCH ITT_ARCH_PPC64
+#  elif defined __loongarch__
+#    define ITT_ARCH ITT_ARCH_LOONGARCH64
 #  endif
 #endif
 

