diff --git a/src/mainwindow.py b/src/mainwindow.py
index 4745427c..29589ff0 100644
--- a/src/mainwindow.py
+++ b/src/mainwindow.py
@@ -54,11 +54,6 @@ try:
 except ImportError as importError:
     NOTFOUND.append('Please install PyQt5: %s' % importError)
 
-try:
-    from zope.interface import implements  # pylint: disable=unused-import
-except ImportError as importError:
-    NOTFOUND.append('Package python-zope-interface missing: %s' % importError)
-
 
 try:
     from twisted.spread import pb # pylint: disable=unused-import
diff --git a/src/scene.py b/src/scene.py
index 8cc62256..5f530c56 100644
--- a/src/scene.py
+++ b/src/scene.py
@@ -7,8 +7,6 @@ SPDX-License-Identifier: GPL-2.0
 
 """
 
-from zope.interface import implements  # pylint: disable=unused-import
-
 from twisted.internet.defer import succeed
 
 from log import logDebug, id4
