--- a/mconfig
+++ b/mconfig
@@ -48,7 +48,7 @@ cflags_opts="-Wall -Werror -Wfatal-errors -Wno-unknown-warning-option \
 -Wtype-limits -Wunused-parameter -Wunused-but-set-parameter \
 -Wno-discarded-qualifiers -Wno-incompatible-pointer-types \
 -pipe -fmessage-length=0 -fPIC"
-cflags="$CFLAGS -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -fstack-protector --param ssp-buffer-size=4"
+cflags="$CFLAGS -Wformat -Wformat-security -fstack-protector --param ssp-buffer-size=4"
 ldflags=$LDFLAGS
 
 package_name=apptainer
--- a/mlocal/frags/debug_opts.mk
+++ b/mlocal/frags/debug_opts.mk
@@ -1,2 +1,2 @@
-CFLAGS := $(filter-out -D_FORTIFY_SOURCE=2 -O2,$(CFLAGS)) -O0 -ggdb
+CFLAGS := $(filter-out -O2,$(CFLAGS)) -O0 -ggdb
 CPPFLAGS += -DDBG
--- a/mlocal/frags/go_runtime_opts.mk
+++ b/mlocal/frags/go_runtime_opts.mk
@@ -1 +1 @@
-CGO_CFLAGS += -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -fstack-protector --param ssp-buffer-size=4
+CGO_CFLAGS += -O2 -Wformat -Wformat-security -fstack-protector --param ssp-buffer-size=4
