diff -ruN a/pilkit/processors/resize.py b/pilkit/processors/resize.py
--- a/pilkit/processors/resize.py	2017-02-17 12:45:57.000000000 +0100
+++ b/pilkit/processors/resize.py	2023-07-06 11:28:56.842558401 +0200
@@ -21,7 +21,7 @@
     def process(self, img):
         if self.upscale or (self.width < img.size[0] and self.height < img.size[1]):
             img = img.convert('RGBA')
-            img = img.resize((self.width, self.height), Image.ANTIALIAS)
+            img = img.resize((self.width, self.height), Image.LANCZOS)
         return img
 
 
