Index: acpica-unix-20211217/source/components/utilities/utdebug.c
===================================================================
--- acpica-unix-20211217.orig/source/components/utilities/utdebug.c
+++ acpica-unix-20211217/source/components/utilities/utdebug.c
@@ -182,7 +182,7 @@ void
 AcpiUtInitStackPtrTrace (
     void)
 {
-    ACPI_SIZE               CurrentSp;
+    static ACPI_SIZE               CurrentSp;
 
 
     AcpiGbl_EntryStackPointer = &CurrentSp;
