Index: acpica-unix-20210331/source/include/platform/acefi.h
===================================================================
--- acpica-unix-20210331.orig/source/include/platform/acefi.h
+++ acpica-unix-20210331/source/include/platform/acefi.h
@@ -173,7 +173,7 @@
 
 #endif
 
-#if defined(__x86_64__)
+#if defined(__x86_64__) || defined(__loongarch64)
 #if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 7))
 #define USE_MS_ABI 1
 #endif
@@ -189,11 +189,11 @@
 
 #define VOID        void
 
-#if defined(__ia64__) || defined(__x86_64__)
+#if defined(__ia64__) || defined(__x86_64__) || defined(__loongarch64)
 
 #define ACPI_MACHINE_WIDTH          64
 
-#if defined(__x86_64__)
+#if defined(__x86_64__) || defined(__loongarch64)
 
 /* for x86_64, EFI_FUNCTION_WRAPPER must be defined */
 
Index: acpica-unix-20210331/source/include/platform/aclinux.h
===================================================================
--- acpica-unix-20210331.orig/source/include/platform/aclinux.h
+++ acpica-unix-20210331/source/include/platform/aclinux.h
@@ -325,6 +325,7 @@
 #if defined(__ia64__)    || (defined(__x86_64__) && !defined(__ILP32__)) ||\
     defined(__aarch64__) || defined(__PPC64__) ||\
     defined(__s390x__) ||\
+    defined(__loongarch64) ||\
     (defined(__riscv) && (defined(__LP64__) || defined(_LP64)))
 #define ACPI_MACHINE_WIDTH          64
 #define COMPILER_DEPENDENT_INT64    long
