--- a/build.sh	2023-04-20 17:19:57.567640306 +0800
+++ b/build.sh	2023-04-20 17:20:56.920551226 +0800
@@ -140,6 +140,7 @@
     ${CONF_OPTS} \
     --enable-debug \
     --disable-optimize \
+    --disable-new-pass-manager \
     --enable-gczeal
   ${MAKE} ${MAKE_OPTS}
   cd ..
@@ -151,6 +152,7 @@
   LLVM_OBJDUMP="${LLVM_OBJDUMP}" \
   ${CONF_OPTS} \
+  --disable-new-pass-manager \
   --enable-optimize
 ${MAKE} ${MAKE_OPTS}
 cd ..
 
