diff --git a/Makefile b/Makefile
index a369415b0..632744313 100644
--- a/Makefile
+++ b/Makefile
@@ -106,8 +106,10 @@ endif
 
 # haxelib should depends on haxe, but we don't want to do that...
 haxelib:
-	(cd $(CURDIR)/extra/haxelib_src && $(CURDIR)/$(HAXE_OUTPUT) client.hxml && nekotools boot run.n)
-	mv extra/haxelib_src/run$(EXTENSION) $(HAXELIB_OUTPUT)
+	cd $(CURDIR)/extra/haxelib_src && \
+	HAXE_STD_PATH=$(CURDIR)/std $(CURDIR)/$(HAXE_OUTPUT) client.hxml && \
+	nekotools boot -c run.n
+	${CC} $(CURDIR)/extra/haxelib_src/run.c -o $(HAXELIB_OUTPUT) -lneko
 
 tools: haxelib
 
