Index: VTK-9.2.6/Utilities/KWIML/vtkkwiml/include/kwiml/abi.h
===================================================================
--- VTK-9.2.6.orig/Utilities/KWIML/vtkkwiml/include/kwiml/abi.h
+++ VTK-9.2.6/Utilities/KWIML/vtkkwiml/include/kwiml/abi.h
@@ -471,6 +471,10 @@ suppression macro KWIML_ABI_NO_VERIFY wa
 #elif defined(__riscv) || defined(__riscv__)
 # define KWIML_ABI_ENDIAN_ID KWIML_ABI_ENDIAN_ID_LITTLE
 
+/* LoongArch */
+#elif defined(__loongarch__)
+# define KWIML_ABI_ENDIAN_ID KWIML_ABI_ENDIAN_ID_LITTLE
+
 /* Unknown CPU */
 #elif !defined(KWIML_ABI_NO_ERROR_ENDIAN)
 # error "Byte order of target CPU unknown."
