--- a/tensorboard/compat/tensorflow_stub/dtypes.py	2024-06-30 15:59:21.290476559 +0300
+++ b/tensorboard/compat/tensorflow_stub/dtypes.py	2024-06-30 15:59:29.377199428 +0300
@@ -674,7 +674,7 @@
         # dtype with a single constant (np.string does not exist) to decide
         # dtype is a "string" type. We need to compare the dtype.type to be
         # sure it's a string type.
-        if type_value.type == np.string_ or type_value.type == np.unicode_:
+        if type_value.type == np.bytes_ or type_value.type == np.str_:
             return string
 
     if isinstance(type_value, (type, np.dtype)):
--- a/tensorboard/util/tensor_util.py	2024-06-30 15:58:32.200133798 +0300
+++ w/tensorboard/util/tensor_util.py	2024-06-30 15:58:50.533595467 +0300
@@ -137,7 +137,7 @@
     # dtype with a single constant (np.string does not exist) to decide
     # dtype is a "string" type. We need to compare the dtype.type to be
     # sure it's a string type.
-    if dtype.type == np.string_ or dtype.type == np.unicode_:
+    if dtype.type == np.bytes_ or dtype.type == np.str_:
         return SlowAppendObjectArrayToTensorProto
     return GetFromNumpyDTypeDict(_NP_TO_APPEND_FN, dtype)
 
