diff --git a/src/meson.build b/src/meson.build
index dddafe9c2..77bd54d16 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -320,7 +320,7 @@ hb_headers = hb_base_headers
 
 harfbuzz_deps = [thread_dep, m_dep] + harfbuzz_extra_deps
 
-libharfbuzz_link_language = 'c'
+libharfbuzz_link_language = 'cpp'
 
 if conf.get('HAVE_FREETYPE', 0) == 1
   hb_sources += hb_ft_sources
@@ -442,7 +442,7 @@ libharfbuzz_subset = library('harfbuzz-subset', hb_subset_sources,
   version: version,
   install: true,
   darwin_versions: darwin_versions,
-  link_language: 'c',
+  link_language: 'cpp',
 )
 
 libharfbuzz_subset_dep = declare_dependency(
