diff --git a/setup.py b/setup.py
index b4688553..4f9372f4 100644
--- a/setup.py
+++ b/setup.py
@@ -2,8 +2,8 @@
 import codecs
 import os
 import sys
-from distutils.util import convert_path
 from fnmatch import fnmatchcase
+from pathlib import Path
 
 from setuptools import find_packages, setup
 
@@ -84,7 +84,7 @@ def find_package_data(
     """
 
     out = {}
-    stack = [(convert_path(where), "", package, only_in_packages)]
+    stack = [(str(Path(where)), "", package, only_in_packages)]
     while stack:
         where, prefix, package, only_in_packages = stack.pop(0)
         for name in os.listdir(where):
@@ -160,6 +160,7 @@ def find_package_data(
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
         "Topic :: Internet :: WWW/HTTP",
     ],
     zip_safe=False,
