Index: sbsigntools/src/coff/pe.h
===================================================================
--- sbsigntools.orig/src/coff/pe.h
+++ sbsigntools/src/coff/pe.h
@@ -152,6 +152,7 @@
 #define IMAGE_FILE_MACHINE_TRICORE           0x0520
 #define IMAGE_FILE_MACHINE_WCEMIPSV2         0x0169
 #define IMAGE_FILE_MACHINE_AARCH64           0xaa64
+#define IMAGE_FILE_MACHINE_LOONGARCH64       0x6264
 #define IMAGE_FILE_MACHINE_RISCV64           0x5064
 
 #define IMAGE_SUBSYSTEM_UNKNOWN			 0
Index: sbsigntools/src/image.c
===================================================================
--- sbsigntools.orig/src/image.c
+++ sbsigntools/src/image.c
@@ -239,6 +239,7 @@ static int image_pecoff_parse(struct ima
 	switch (magic) {
 	case IMAGE_FILE_MACHINE_AMD64:
 	case IMAGE_FILE_MACHINE_AARCH64:
+	case IMAGE_FILE_MACHINE_LOONGARCH64:
 	case IMAGE_FILE_MACHINE_RISCV64:
 		rc = image_pecoff_parse_64(image);
 		break;
