diff --git a/meson.build b/meson.build
index 9ae278b66..91fad3dbc 100644
--- a/meson.build
+++ b/meson.build
@@ -430,6 +430,8 @@ if build_standalone and efiboot.found() and efivar.found()
           EFI_MACHINE_TYPE_NAME = 'arm'
   elif host_cpu == 'aarch64'
           EFI_MACHINE_TYPE_NAME = 'aa64'
+  elif host_cpu == 'loongarch64'
+          EFI_MACHINE_TYPE_NAME = 'loongarch64'
   else
           EFI_MACHINE_TYPE_NAME = ''
   endif
diff --git a/plugins/uefi-capsule/fu-uefi-common.c b/plugins/uefi-capsule/fu-uefi-common.c
index e501c54ef..39dfc7328 100644
--- a/plugins/uefi-capsule/fu-uefi-common.c
+++ b/plugins/uefi-capsule/fu-uefi-common.c
@@ -26,6 +26,8 @@ fu_uefi_bootmgr_get_suffix(GError **error)
 		{64, "x64"},
 #elif defined(__aarch64__)
 		{64, "aa64"},
+#elif defined(__loongarch64)
+		{64, "loongarch64"},
 #endif
 #if defined(__x86_64__) || defined(__i386__) || defined(__i686__)
 		{32, "ia32"},
