Index: htmlcxx-0.87/html/CharsetConverter.h
===================================================================
--- htmlcxx-0.87.orig/html/CharsetConverter.h
+++ htmlcxx-0.87/html/CharsetConverter.h
@@ -17,7 +17,7 @@ namespace htmlcxx
 						: std::runtime_error(arg) {}
 			};
 			
-			CharsetConverter(const std::string &from, const std::string &to) throw (Exception);
+			CharsetConverter(const std::string &from, const std::string &to) noexcept(false);
 			~CharsetConverter();
 			
 			std::string convert(const std::string &input);
Index: htmlcxx-0.87/html/CharsetConverter.cc
===================================================================
--- htmlcxx-0.87.orig/html/CharsetConverter.cc
+++ htmlcxx-0.87/html/CharsetConverter.cc
@@ -7,7 +7,7 @@
 using namespace std;
 using namespace htmlcxx;
 
-CharsetConverter::CharsetConverter(const string &from, const string &to) throw (Exception)
+CharsetConverter::CharsetConverter(const string &from, const string &to) noexcept(false)
 {
 	mIconvDescriptor = iconv_open(to.c_str(), from.c_str());
 	if (mIconvDescriptor == (iconv_t)(-1))
