diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6195c3b..1c227d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,14 +13,14 @@ find_package(PkgConfig REQUIRED)
 find_package(Qt5 REQUIRED Core Widgets Network DBus Gui LinguistTools)
 find_package(Dtk REQUIRED Widget Gui)
 find_package(PackageKitQt5 REQUIRED)
-find_package(AppStreamQt REQUIRED)
+find_package(AppStreamQt5 REQUIRED)
 find_package(PkgConfig REQUIRED)
 include(FindPkgConfig)
 include(GNUInstallDirs)
 
 include_directories(
     ${PackageKitQt5_INCLUDE_DIR}
-    ${AppStreamQt_INCLUDE_DIRS}
+    ${AppStreamQt5_INCLUDE_DIRS}
 )
 
 file(GLOB TS_FILES ./translations/*.ts)
@@ -63,7 +63,7 @@ add_executable(${PROJECT_NAME}
 
 target_include_directories(${PROJECT_NAME} PUBLIC
     ${PackageKitQt5_INCLUDE_DIR}
-    ${AppStreamQt_INCLUDE_DIRS}
+    ${AppStreamQt5_INCLUDE_DIRS}
 )
 
 target_link_libraries(${PROJECT_NAME}
@@ -82,4 +82,4 @@ set(CMAKE_INSTALL_PREFIX /usr)
 
 install(TARGETS dde-store DESTINATION bin)
 install(FILES ${QM_FILES} DESTINATION share/dde-store/translations)
-install(FILES resources/dde-store.desktop DESTINATION share/applications)
\ No newline at end of file
+install(FILES resources/dde-store.desktop DESTINATION share/applications)
diff --git a/backend/sources/packagekit/packagekitsource.cpp b/backend/sources/packagekit/packagekitsource.cpp
index 27bb5b7..2c69de9 100644
--- a/backend/sources/packagekit/packagekitsource.cpp
+++ b/backend/sources/packagekit/packagekitsource.cpp
@@ -2,10 +2,11 @@
 #include "backend/settings.h"
 #include "backend/ratingshelper.h"
 #include <Details>
-#include <AppStreamQt/pool.h>
-#include <AppStreamQt/icon.h>
-#include <AppStreamQt/screenshot.h>
-#include <AppStreamQt/image.h>
+#include <AppStreamQt5/developer.h>
+#include <AppStreamQt5/pool.h>
+#include <AppStreamQt5/icon.h>
+#include <AppStreamQt5/screenshot.h>
+#include <AppStreamQt5/image.h>
 
 using namespace PackageKit;
 using namespace AppStream;
@@ -103,10 +104,10 @@ void PackageKitSource::getFullData(App *app)
     if (metadata.contains(app->package)) {
         const Component data = metadata.value(app->package);
 
-        app->developer = data.developerName();
+        app->developer = data.developer().name();
         app->description = data.description();
         
-        for (Screenshot screenshot : data.screenshots()) {
+        for (Screenshot screenshot : data.screenshotsAll()) {
             for (Image image : screenshot.images()) {
                 if (image.kind() == Image::KindSource)
                     app->screenshots << image.url();
diff --git a/backend/sources/packagekit/packagekitsource.h b/backend/sources/packagekit/packagekitsource.h
index 3abb5a1..b63dd2e 100644
--- a/backend/sources/packagekit/packagekitsource.h
+++ b/backend/sources/packagekit/packagekitsource.h
@@ -3,7 +3,7 @@
 
 #include "backend/sourcemanager.h"
 #include <Daemon>
-#include <AppStreamQt/component.h>
+#include <AppStreamQt5/component.h>
 
 class PackageKitSource : public Source
 {
@@ -36,4 +36,4 @@ private:
     QList<App*> ratingsQueue;
 };
 
-#endif // PACKAGEKITSOURCE_H
\ No newline at end of file
+#endif // PACKAGEKITSOURCE_H
