diff --git tests/test_awslambda/test_lambda_layers_invoked.py tests/test_awslambda/test_lambda_layers_invoked.py
index fc181e5a3..49e2971a7 100644
--- tests/test_awslambda/test_lambda_layers_invoked.py
+++ tests/test_awslambda/test_lambda_layers_invoked.py
@@ -5,6 +5,7 @@ from moto import mock_lambda
 from uuid import uuid4
 
 from .utilities import get_role_name, _process_lambda
+from ..markers import requires_docker
 
 PYTHON_VERSION = "python3.11"
 _lambda_region = "us-west-2"
@@ -20,6 +21,7 @@ def lambda_handler(event, context):
     return _process_lambda(pfunc)
 
 
+@requires_docker
 @mock_lambda
 def test_invoke_local_lambda_layers():
     conn = boto3.client("lambda", _lambda_region)
