diff -ur moto-4.2.2.orig/moto/identitystore/models.py moto-4.2.2/moto/identitystore/models.py
--- moto-4.2.2.orig/moto/identitystore/models.py	2023-09-08 20:34:39.481595401 +0800
+++ moto-4.2.2/moto/identitystore/models.py	2023-09-08 20:35:43.999914179 +0800
@@ -1,5 +1,10 @@
-from typing import Dict, Tuple, List, Any, NamedTuple, Optional
-from typing_extensions import Self
+from typing import Dict, Tuple, List, Any, NamedTuple, Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+else:
+    Self = object
+
 from moto.utilities.paginator import paginate
 
 from botocore.exceptions import ParamValidationError
