diff --git a/configure b/configure
index 4b7f993..7254086 100755
--- a/configure
+++ b/configure
@@ -23070,7 +23070,7 @@ $as_echo_n "checking for $am_display_PYTHON version... " >&6; }
 if ${am_cv_python_version+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  am_cv_python_version=`$PYTHON -c "import sys; sys.stdout.write(sys.version[:3])"`
+  am_cv_python_version=`$PYTHON -c "import sys; sys.stdout.write(sys.version[:4])"`
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $am_cv_python_version" >&5
 $as_echo "$am_cv_python_version" >&6; }
@@ -23111,7 +23111,7 @@ else:
 # <https://github.com/pypa/virtualenv/issues/118>
 try:
     from platform import python_implementation
-    if python_implementation() == 'CPython' and sys.version[:3] == '2.7':
+    if python_implementation() == 'CPython' and sys.version[:4] == '2.7':
         can_use_sysconfig = 0
 except ImportError:
     pass"
diff --git a/src/bin/shell/Makefile.in b/src/bin/shell/Makefile.in
index 053684d..8af1b82 100644
--- a/src/bin/shell/Makefile.in
+++ b/src/bin/shell/Makefile.in
@@ -534,7 +534,6 @@ install-pkgpythonPYTHON: $(pkgpython_PYTHON)
 	  if test -f "$$p"; then b=; else b="$(srcdir)/"; fi; \
 	  if test -f $$b$$p; then \
 	    $(am__strip_dir) \
-	    dlist="$$dlist $$f"; \
 	    list2="$$list2 $$b$$p"; \
 	  else :; fi; \
 	done; \
