Index: android-tools-34.0.1/vendor/boringssl/include/openssl/base.h
===================================================================
--- android-tools-34.0.1.orig/vendor/boringssl/include/openssl/base.h
+++ android-tools-34.0.1/vendor/boringssl/include/openssl/base.h
@@ -118,6 +118,8 @@ extern "C" {
 #define OPENSSL_32_BIT
 #elif defined(__myriad2__)
 #define OPENSSL_32_BIT
+#elif defined(__loongarch_lp64)
+#define OPENSSL_64_BIT
 #else
 // Note BoringSSL only supports standard 32-bit and 64-bit two's-complement,
 // little-endian architectures. Functions will not produce the correct answer
