--- android-tools-35.0.1/vendor/boringssl/include/openssl/target.h	2024-09-19 19:56:04.698232853 +0800
+++ android-tools-35.0.1/vendor/boringssl/include/openssl/target.h	2024-09-19 19:56:59.126341496 +0800
@@ -34,6 +34,9 @@
 #elif defined(__ARMEL__) || defined(_M_ARM)
 #define OPENSSL_32_BIT
 #define OPENSSL_ARM
+#elif defined(__loongarch_lp64)
+#define OPENSSL_64_BIT
+#define OPENSSL_LOONGARCH
 #elif defined(__MIPSEL__) && !defined(__LP64__)
 #define OPENSSL_32_BIT
 #define OPENSSL_MIPS
