diff -ruN a/remoto/tests/backends/test_backends.py b/remoto/tests/backends/test_backends.py
--- a/remoto/tests/backends/test_backends.py	2021-05-18 21:58:12.000000000 +0200
+++ b/remoto/tests/backends/test_backends.py	2024-04-14 11:23:42.644557039 +0200
@@ -1,5 +1,5 @@
 import sys
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 import pytest
 from remoto import backends
 from remoto.backends import local
diff -ruN a/remoto/tests/test_log.py b/remoto/tests/test_log.py
--- a/remoto/tests/test_log.py	2021-05-18 21:58:12.000000000 +0200
+++ b/remoto/tests/test_log.py	2024-04-14 11:23:11.144417753 +0200
@@ -1,7 +1,7 @@
 from pytest import raises
 from remoto import log
 from remoto.exc import TimeoutError
-from mock import Mock
+from unittest.mock import Mock
 
 
 class TestReporting(object):
diff -ruN a/remoto/tests/test_process.py b/remoto/tests/test_process.py
--- a/remoto/tests/test_process.py	2021-05-18 21:58:12.000000000 +0200
+++ b/remoto/tests/test_process.py	2024-04-14 11:22:54.014342103 +0200
@@ -1,4 +1,4 @@
-from mock import Mock
+from unittest.mock import Mock
 from remoto import process
 
 
diff -ruN a/remoto/tests/test_rsync.py b/remoto/tests/test_rsync.py
--- a/remoto/tests/test_rsync.py	2021-05-18 21:58:12.000000000 +0200
+++ b/remoto/tests/test_rsync.py	2024-04-14 11:22:32.607580968 +0200
@@ -1,4 +1,4 @@
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 from remoto import file_sync
 
 
