Index: jemalloc-5.2.1/include/jemalloc/internal/quantum.h
===================================================================
--- jemalloc-5.2.1.orig/include/jemalloc/internal/quantum.h
+++ jemalloc-5.2.1/include/jemalloc/internal/quantum.h
@@ -30,6 +30,9 @@
 #  ifdef __hppa__
 #    define LG_QUANTUM		4
 #  endif
+#  ifdef __loongarch64
+#    define LG_QUANTUM		4
+#  endif
 #  ifdef __m68k__
 #    define LG_QUANTUM		3
 #  endif
