From 12675269a860728adea9c962ba67d613376dc397 Mon Sep 17 00:00:00 2001
From: Charles Gagnon <charlesg99@outlook.com>
Date: Wed, 3 Apr 2024 21:01:02 -0400
Subject: [PATCH] Fix added panel menu button detection

---
 panelManager.js | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/panelManager.js b/panelManager.js
index c76a2720..917c91dd 100755
--- a/panelManager.js
+++ b/panelManager.js
@@ -202,9 +202,11 @@ export const PanelManager = class {
             [
                 Main.panel[c], 
                 'child-added', 
-                (parent, child) => 
-                    this.primaryPanel && 
-                    this._adjustPanelMenuButton(this._getPanelMenuButton(child), this.primaryPanel.monitor, this.primaryPanel.getPosition())
+                (parent, child) => {
+                    this.primaryPanel &&
+                    child instanceof St.Bin && 
+                    this._adjustPanelMenuButton(this._getPanelMenuButton(child.get_first_child()), this.primaryPanel.monitor, this.primaryPanel.getPosition())
+                }
             ]
         ));
 
