From f281c94341de06d77672fe5cb17b57ec1217c651 Mon Sep 17 00:00:00 2001
From: Proxtx <gopro54122@gmail.com>
Date: Sat, 14 Oct 2023 03:13:07 +0200
Subject: [PATCH] Fix crash on doubletapping close button in window preview

---
 windowPreview.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/windowPreview.js b/windowPreview.js
index d9c1d670..6dcff6dd 100644
--- a/windowPreview.js
+++ b/windowPreview.js
@@ -877,6 +877,7 @@ export const Preview = GObject.registerClass({
     }
 
     _onCloseBtnClick() {
+        if(!this.reactive) return;
         this._hideOrShowCloseButton(true);
         this.reactive = false;
 
