diff -up dash-to-panel-55/Makefile.orig dash-to-panel-55/Makefile
--- dash-to-panel-55/Makefile.orig	2023-03-09 16:32:54.000000000 +0100
+++ dash-to-panel-55/Makefile	2023-03-15 13:42:54.227368912 +0100
@@ -10,9 +10,12 @@ EXTRA_IMAGES = highlight_stacked_bg.svg
 TOLOCALIZE =  prefs.js appIcons.js
 MSGSRC = $(wildcard po/*.po)
 ifeq ($(strip $(DESTDIR)),)
+	INSTALLTYPE = local
 	INSTALLBASE = $(HOME)/.local/share/gnome-shell/extensions
 else
+	INSTALLTYPE = system
 	INSTALLBASE = $(DESTDIR)/usr/share/gnome-shell/extensions
+	SHARE_PREFIX = $(DESTDIR)/usr/share
 endif
 INSTALLNAME = dash-to-panel@jderose9.github.com
 
@@ -71,6 +74,12 @@ install-local: _build
 	rm -rf $(INSTALLBASE)/$(INSTALLNAME)
 	mkdir -p $(INSTALLBASE)/$(INSTALLNAME)
 	cp -r ./_build/* $(INSTALLBASE)/$(INSTALLNAME)/
+ifeq ($(INSTALLTYPE),system)
+	rm -r $(INSTALLBASE)/$(INSTALLNAME)/schemas $(INSTALLBASE)/$(INSTALLNAME)/locale
+	mkdir -p $(SHARE_PREFIX)/glib-2.0/schemas $(SHARE_PREFIX)/locale
+	cp -r ./schemas/*gschema.* $(SHARE_PREFIX)/glib-2.0/schemas
+	cp -r ./_build/locale/* $(SHARE_PREFIX)/locale
+endif
 	-rm -fR _build
 	echo done
 
