diff --git a/src/sage_docbuild/sphinxbuild.py b/src/sage_docbuild/sphinxbuild.py
index 94407cad5eb..ffd422bc026 100644
--- a/src/sage_docbuild/sphinxbuild.py
+++ b/src/sage_docbuild/sphinxbuild.py
@@ -109,6 +109,7 @@ class SageSphinxLogger():
         # We fail whenever a line starts with "WARNING:", however, we ignore
         # these warnings, as they are not relevant.
         self._ignored_warnings = (
+            re.compile('WARNING: could not parse arglist'),
             re.compile("WARNING: favicon file 'favicon.ico' does not exist"),
             re.compile('WARNING: html_static_path entry .* does not exist'),
             re.compile('WARNING: while setting up extension'),
