diff --git i/fire/core_test.py w/fire/core_test.py
index 75b7699..0f14d41 100644
--- i/fire/core_test.py
+++ w/fire/core_test.py
@@ -22,7 +22,7 @@ from fire import core
 from fire import test_components as tc
 from fire import testutils
 from fire import trace
-import mock
+from unittest import mock
 
 import six
 
diff --git i/fire/fire_import_test.py w/fire/fire_import_test.py
index c597568..b42be22 100644
--- i/fire/fire_import_test.py
+++ w/fire/fire_import_test.py
@@ -18,7 +18,7 @@ import sys
 
 import fire
 from fire import testutils
-import mock
+from unittest import mock
 
 
 class FireImportTest(testutils.BaseTestCase):
diff --git i/fire/fire_test.py w/fire/fire_test.py
index 8b904c2..886f57e 100644
--- i/fire/fire_test.py
+++ w/fire/fire_test.py
@@ -25,7 +25,7 @@ import fire
 from fire import test_components as tc
 from fire import testutils
 
-import mock
+from unittest import mock
 import six
 
 
diff --git i/fire/interact_test.py w/fire/interact_test.py
index 29fa759..922f02d 100644
--- i/fire/interact_test.py
+++ w/fire/interact_test.py
@@ -21,7 +21,7 @@ from __future__ import print_function
 from fire import interact
 from fire import testutils
 
-import mock
+from unittest import mock
 
 
 try:
diff --git i/fire/testutils.py w/fire/testutils.py
index 5f87514..65a01dd 100644
--- i/fire/testutils.py
+++ w/fire/testutils.py
@@ -27,7 +27,7 @@ import unittest
 from fire import core
 from fire import trace
 
-import mock
+from unittest import mock
 import six
 
 
