diff --git a/data/meson.build b/data/meson.build
index 8142127..a1e5c96 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -22,7 +22,6 @@ install_data ('smooth.png', install_dir: join_paths (pkgdatadir, 'themes'))
 install_data ('edu_kang_xi.png', install_dir: join_paths (pkgdatadir, 'themes'))
 
 desktop_file = i18n.merge_file (
-  'desktop-file',
   input:       'org.gnome.Mahjongg.desktop.in',
   output:      'org.gnome.Mahjongg.desktop',
   install:     true,
@@ -42,7 +41,7 @@ if desktop_file_validate.found()
   )
 endif
 
-appdata_file = i18n.merge_file ('appdata-file',
+appdata_file = i18n.merge_file (
                  input: 'org.gnome.Mahjongg.appdata.xml.in',
                  output: 'org.gnome.Mahjongg.appdata.xml',
                  install: true,
