--- qt-creator-opensource-src-9.0.0/src/plugins/clangformat/clangformatutils.cpp	2023-02-23 10:51:37.220677845 +0800
+++ qt-creator-opensource-src-9.0.0/src/plugins/clangformat/clangformatutils.cpp	2023-02-23 10:51:30.321059140 +0800
@@ -147,7 +147,7 @@
 #else
     style.SortIncludes = true;
 #endif
-    style.SortUsingDeclarations = true;
+    style.SortUsingDeclarations = FormatStyle::SUD_LexicographicNumeric;
     style.SpaceAfterCStyleCast = true;
     style.SpaceAfterTemplateKeyword = false;
     style.SpaceBeforeAssignmentOperators = true;
--- qt-creator-opensource-src-9.0.0/src/plugins/clangformat/clangformatbaseindenter.cpp	2023-02-23 10:52:38.537288962 +0800
+++ qt-creator-opensource-src-9.0.0/src/plugins/clangformat/clangformatbaseindenter.cpp	2023-02-23 10:52:34.713500314 +0800
@@ -35,7 +35,7 @@
 #else
     style.SortIncludes = false;
 #endif
-    style.SortUsingDeclarations = false;
+    style.SortUsingDeclarations = clang::format::FormatStyle::SUD_Never;
 
     // This is a separate pass, don't do it unless it's the full formatting.
     style.FixNamespaceComments = false;
