diff --git i/imagick.c w/imagick.c
index 7733e2b..1533b15 100644
--- i/imagick.c
+++ w/imagick.c
@@ -535,7 +535,7 @@ static void php_imagick_init_globals(zend_imagick_globals *imagick_globals)
 {
 	imagick_globals->locale_fix = 0;
 	imagick_globals->progress_monitor = 0;
-	imagick_globals->skip_version_check = 0;
+	imagick_globals->skip_version_check = 1;
 	imagick_globals->set_single_thread = 1;
 	imagick_globals->allow_zero_dimension_images = 0;
 	// 10 is magick number, that seems to be enough.
