diff --git a/src/private/dquickkeylistener.cpp b/src/private/dquickkeylistener.cpp
index 5d2c733..4f8699b 100644
--- a/src/private/dquickkeylistener.cpp
+++ b/src/private/dquickkeylistener.cpp
@@ -146,7 +146,7 @@ int DQuickKeyListenerPrivate::doNativeShiftKey(QKeyEvent *e, int nativeKey)
     if (!(e->modifiers() & Qt::ShiftModifier))
         return key;
 
-    QList<int> possibleKeys = QKeyMapper::possibleKeys(e);
+    QList<QKeyCombination> possibleKeys = QKeyMapper::possibleKeys(e);
     int pkTotal = possibleKeys.count();
     if (!pkTotal)
         return false;
