From b703c0e025be6a080e49e9e14a71ae9af4c585c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=A1clav=20=C5=A0milauer?= <eu@doxos.eu>
Date: Sun, 2 Apr 2023 10:32:59 +0200
Subject: [PATCH] Fix Tesseract abbreviation for Czech (ces, not cse)

---
 src/ocrfeeder/util/configuration.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ocrfeeder/util/configuration.py b/src/ocrfeeder/util/configuration.py
index 1eec7f5..ef9a065 100644
--- a/src/ocrfeeder/util/configuration.py
+++ b/src/ocrfeeder/util/configuration.py
@@ -39,7 +39,7 @@ PREDEFINED_ENGINES = {'tesseract': {'name': 'Tesseract',
                                        '$FILE.txt; rm $FILE $FILE.txt'],
                                     'language_argument': '-l',
                                     'languages': 'af:afr,ar:ara,az:aze,be:bel,'
-                                       'bn:ben,bg:bul,ca:cat,cs:cse,zh:chi-sim,'
+                                       'bn:ben,bg:bul,ca:cat,cs:ces,zh:chi-sim,'
                                        'chr:chr,da:dan,de:deu,el:ell,en:eng,'
                                        'et:est,eu:eus,fi:fin,fr:fra,gl:glg,he:heb,'
                                        'hi:hin,hr:hrv,hu:hun,id:ind,is:isl,it:ita,'
-- 
GitLab

