From 2a580ed065764c96da34634fc9186f81234f8ca5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 7 Jun 2021 11:17:51 +0000
Subject: [PATCH] Port to GooCanvas 3.0

---
 src/ocrfeeder/studio/studioBuilder.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ocrfeeder/studio/studioBuilder.py b/src/ocrfeeder/studio/studioBuilder.py
index 7a2ccdc..e8e3274 100644
--- a/src/ocrfeeder/studio/studioBuilder.py
+++ b/src/ocrfeeder/studio/studioBuilder.py
@@ -19,7 +19,7 @@
 
 import gi
 gi.require_version('Gtk', '3.0')
-gi.require_version('GooCanvas', '2.0')
+gi.require_version('GooCanvas', '3.0')
 gi.require_version('GtkSpell', '3.0')
 from ocrfeeder.util import lib
 from ocrfeeder.util.constants import *
-- 
GitLab

