Index: sbc/sbc/sbc_primitives.c
===================================================================
--- sbc.orig/sbc/sbc_primitives.c
+++ sbc/sbc/sbc_primitives.c
@@ -593,7 +593,9 @@ static int sbc_calc_scalefactors_j(
 
 static void sbc_init_primitives_x86(struct sbc_encoder_state *state)
 {
+#ifdef x86_64
 	__builtin_cpu_init();
+#endif
 
 #ifdef SBC_BUILD_WITH_MMX_SUPPORT
 	if (__builtin_cpu_supports("mmx"))
